{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load data from all_of_plos collection\n",
    "\n",
    "Data is in JATS XML format\n",
    "https://en.wikipedia.org/wiki/Journal_Article_Tag_Suite\n",
    "\n",
    "Nice guide to usage: http://www.thehackerwithin.org/berkeley/plos.html\n",
    "This project could potentially also be helpful: https://github.com/titipata/pubmed_parser\n",
    "Another potentially useful repo: https://github.com/elifesciences/jats-scraper\n",
    "\n",
    "GETTING STARTED:\n",
    "Just download the entire collection of xml files (>10GB)\n",
    "\n",
    "Alternatively, supposedly you can use pip via the official repo (but it didn't work for me):\n",
    "official repo: https://github.com/PLOS/allofplos\n",
    "Install All Of Plos:        pip install allofplos                         \n",
    "                            python -m allofplos.update              (installs the >10GB of article files)\n",
    "               "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import random\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "path2directory = '/home/brch/Data/allofplos/'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "254570\n"
     ]
    }
   ],
   "source": [
    "# load all the xml filenames from the directory path into a list\n",
    "#   os.listdir won't work on mac (probably)...you can use the glob package instead (glob.glob)\n",
    "\n",
    "xml_list = os.listdir(path2directory)\n",
    "\n",
    "L = len(xml_list)\n",
    "print(L) # ~250,000 articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "document number 166067\n",
      "PATH: /home/brch/Data/allofplos/journal.pone.0082393.xml\n",
      "CONTENTS:\n",
      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
      "<!DOCTYPE article PUBLIC \"-//NLM//DTD Journal Publishing DTD v3.0 20080202//EN\" \"http://dtd.nlm.nih.gov/publishing/3.0/journalpublishing3.dtd\">\n",
      "<article article-type=\"research-article\" dtd-version=\"3.0\" xml:lang=\"en\" xmlns:mml=\"http://www.w3.org/1998/Math/MathML\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
      " <front>\n",
      "  <journal-meta>\n",
      "   <journal-id journal-id-type=\"nlm-ta\">\n",
      "    PLoS ONE\n",
      "   </journal-id>\n",
      "   <journal-id journal-id-type=\"publisher-id\">\n",
      "    plos\n",
      "   </journal-id>\n",
      "   <journal-id journal-id-type=\"pmc\">\n",
      "    plosone\n",
      "   </journal-id>\n",
      "   <journal-title-group>\n",
      "    <journal-title>\n",
      "     PLoS ONE\n",
      "    </journal-title>\n",
      "   </journal-title-group>\n",
      "   <issn pub-type=\"epub\">\n",
      "    1932-6203\n",
      "   </issn>\n",
      "   <publisher>\n",
      "    <publisher-name>\n",
      "     Public Library of Science\n",
      "    </publisher-name>\n",
      "    <publisher-loc>\n",
      "     San Francisco, USA\n",
      "    </publisher-loc>\n",
      "   </publisher>\n",
      "  </journal-meta>\n",
      "  <article-meta>\n",
      "   <article-id pub-id-type=\"publisher-id\">\n",
      "    PONE-D-12-15750\n",
      "   </article-id>\n",
      "   <article-id pub-id-type=\"doi\">\n",
      "    10.1371/journal.pone.0082393\n",
      "   </article-id>\n",
      "   <article-categories>\n",
      "    <subj-group subj-group-type=\"heading\">\n",
      "     <subject>\n",
      "      Research Article\n",
      "     </subject>\n",
      "    </subj-group>\n",
      "    <subj-group subj-group-type=\"Discipline-v2\">\n",
      "     <subject>\n",
      "      Biology\n",
      "     </subject>\n",
      "     <subj-group>\n",
      "      <subject>\n",
      "       Computational biology\n",
      "      </subject>\n",
      "      <subj-group>\n",
      "       <subject>\n",
      "        Genomics\n",
      "       </subject>\n",
      "       <subj-group>\n",
      "        <subject>\n",
      "         Genome analysis tools\n",
      "        </subject>\n",
      "        <subj-group>\n",
      "         <subject>\n",
      "          Genetic networks\n",
      "         </subject>\n",
      "        </subj-group>\n",
      "       </subj-group>\n",
      "      </subj-group>\n",
      "      <subj-group>\n",
      "       <subject>\n",
      "        Biological data management\n",
      "       </subject>\n",
      "       <subject>\n",
      "        Regulatory networks\n",
      "       </subject>\n",
      "      </subj-group>\n",
      "     </subj-group>\n",
      "     <subj-group>\n",
      "      <subject>\n",
      "       Genetics\n",
      "      </subject>\n",
      "      <subj-group>\n",
      "       <subject>\n",
      "        Gene function\n",
      "       </subject>\n",
      "       <subject>\n",
      "        Gene networks\n",
      "       </subject>\n",
      "      </subj-group>\n",
      "     </subj-group>\n",
      "     <subj-group>\n",
      "      <subject>\n",
      "       Genomics\n",
      "      </subject>\n",
      "     </subj-group>\n",
      "    </subj-group>\n",
      "    <subj-group subj-group-type=\"Discipline-v2\">\n",
      "     <subject>\n",
      "      Mathematics\n",
      "     </subject>\n",
      "     <subj-group>\n",
      "      <subject>\n",
      "       Applied mathematics\n",
      "      </subject>\n",
      "      <subj-group>\n",
      "       <subject>\n",
      "        Algorithms\n",
      "       </subject>\n",
      "      </subj-group>\n",
      "     </subj-group>\n",
      "     <subj-group>\n",
      "      <subject>\n",
      "       Statistics\n",
      "      </subject>\n",
      "     </subj-group>\n",
      "    </subj-group>\n",
      "   </article-categories>\n",
      "   <title-group>\n",
      "    <article-title>\n",
      "     Inferring Regulatory Networks by Combining Perturbation Screens and Steady State Gene Expression Profiles\n",
      "    </article-title>\n",
      "    <alt-title alt-title-type=\"running-head\">\n",
      "     Inferring Regulatory Networks\n",
      "    </alt-title>\n",
      "   </title-group>\n",
      "   <contrib-group>\n",
      "    <contrib contrib-type=\"author\" equal-contrib=\"yes\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Shojaie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Ali\n",
      "      </given-names>\n",
      "     </name>\n",
      "     <xref ref-type=\"aff\" rid=\"aff1\">\n",
      "      <sup>\n",
      "       1\n",
      "      </sup>\n",
      "     </xref>\n",
      "    </contrib>\n",
      "    <contrib contrib-type=\"author\" equal-contrib=\"yes\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Jauhiainen\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Alexandra\n",
      "      </given-names>\n",
      "     </name>\n",
      "     <xref ref-type=\"aff\" rid=\"aff2\">\n",
      "      <sup>\n",
      "       2\n",
      "      </sup>\n",
      "     </xref>\n",
      "    </contrib>\n",
      "    <contrib contrib-type=\"author\" equal-contrib=\"yes\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kallitsis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Michael\n",
      "      </given-names>\n",
      "     </name>\n",
      "     <xref ref-type=\"aff\" rid=\"aff3\">\n",
      "      <sup>\n",
      "       3\n",
      "      </sup>\n",
      "     </xref>\n",
      "    </contrib>\n",
      "    <contrib contrib-type=\"author\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       George\n",
      "      </given-names>\n",
      "     </name>\n",
      "     <xref ref-type=\"aff\" rid=\"aff3\">\n",
      "      <sup>\n",
      "       3\n",
      "      </sup>\n",
      "     </xref>\n",
      "     <xref ref-type=\"corresp\" rid=\"cor1\">\n",
      "      <sup>\n",
      "       *\n",
      "      </sup>\n",
      "     </xref>\n",
      "    </contrib>\n",
      "   </contrib-group>\n",
      "   <aff id=\"aff1\">\n",
      "    <label>\n",
      "     1\n",
      "    </label>\n",
      "    <addr-line>\n",
      "     Department of Biostatistics, University of Washington, Seattle, Washington, United States of America\n",
      "    </addr-line>\n",
      "   </aff>\n",
      "   <aff id=\"aff2\">\n",
      "    <label>\n",
      "     2\n",
      "    </label>\n",
      "    <addr-line>\n",
      "     Department of Medical Epidemiology and Biostatistics, Karolinska Institutet, Stockholm, Sweden\n",
      "    </addr-line>\n",
      "   </aff>\n",
      "   <aff id=\"aff3\">\n",
      "    <label>\n",
      "     3\n",
      "    </label>\n",
      "    <addr-line>\n",
      "     Department of Statistics, University of Michigan, Ann Arbor, Michigan, United States of America\n",
      "    </addr-line>\n",
      "   </aff>\n",
      "   <contrib-group>\n",
      "    <contrib contrib-type=\"editor\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       de la Fuente\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Alberto\n",
      "      </given-names>\n",
      "     </name>\n",
      "     <role>\n",
      "      Editor\n",
      "     </role>\n",
      "     <xref ref-type=\"aff\" rid=\"edit1\"/>\n",
      "    </contrib>\n",
      "   </contrib-group>\n",
      "   <aff id=\"edit1\">\n",
      "    <addr-line>\n",
      "     Leibniz-Institute for Farm Animal Biology (FBN), Germany\n",
      "    </addr-line>\n",
      "   </aff>\n",
      "   <author-notes>\n",
      "    <corresp id=\"cor1\">\n",
      "     * E-mail:\n",
      "     <email xlink:type=\"simple\">\n",
      "      gmichail@umich.edu\n",
      "     </email>\n",
      "    </corresp>\n",
      "    <fn fn-type=\"conflict\">\n",
      "     <p>\n",
      "      The authors have declared that no competing interests exist.\n",
      "     </p>\n",
      "    </fn>\n",
      "    <fn fn-type=\"con\">\n",
      "     <p>\n",
      "      Conceived and designed the experiments: GM AS. Performed the experiments: AJ MK AS. Analyzed the data: AJ. Contributed reagents/materials/analysis tools: AJ MK GM AS. Wrote the paper: GM AS AJ MK.\n",
      "     </p>\n",
      "    </fn>\n",
      "   </author-notes>\n",
      "   <pub-date pub-type=\"collection\">\n",
      "    <year>\n",
      "     2014\n",
      "    </year>\n",
      "   </pub-date>\n",
      "   <pub-date pub-type=\"epub\">\n",
      "    <day>\n",
      "     28\n",
      "    </day>\n",
      "    <month>\n",
      "     2\n",
      "    </month>\n",
      "    <year>\n",
      "     2014\n",
      "    </year>\n",
      "   </pub-date>\n",
      "   <volume>\n",
      "    9\n",
      "   </volume>\n",
      "   <issue>\n",
      "    2\n",
      "   </issue>\n",
      "   <elocation-id>\n",
      "    e82393\n",
      "   </elocation-id>\n",
      "   <history>\n",
      "    <date date-type=\"received\">\n",
      "     <day>\n",
      "      26\n",
      "     </day>\n",
      "     <month>\n",
      "      5\n",
      "     </month>\n",
      "     <year>\n",
      "      2012\n",
      "     </year>\n",
      "    </date>\n",
      "    <date date-type=\"accepted\">\n",
      "     <day>\n",
      "      1\n",
      "     </day>\n",
      "     <month>\n",
      "      11\n",
      "     </month>\n",
      "     <year>\n",
      "      2013\n",
      "     </year>\n",
      "    </date>\n",
      "   </history>\n",
      "   <permissions>\n",
      "    <copyright-year>\n",
      "     2014\n",
      "    </copyright-year>\n",
      "    <copyright-holder>\n",
      "     Shojaie et al\n",
      "    </copyright-holder>\n",
      "    <license xlink:type=\"simple\">\n",
      "     <license-p>\n",
      "      This is an open-access article distributed under the terms of the\n",
      "      <ext-link ext-link-type=\"uri\" xlink:href=\"http://creativecommons.org/licenses/by/4.0/\" xlink:type=\"simple\">\n",
      "       Creative Commons Attribution License\n",
      "      </ext-link>\n",
      "      , which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.\n",
      "     </license-p>\n",
      "    </license>\n",
      "   </permissions>\n",
      "   <abstract>\n",
      "    <p>\n",
      "     Reconstructing transcriptional regulatory networks is an important task in functional genomics. Data obtained from experiments that perturb genes by knockouts or RNA interference contain useful information for addressing this reconstruction problem. However, such data can be limited in size and/or are expensive to acquire. On the other hand, observational data of the organism in steady state (e.g., wild-type) are more readily available, but their informational content is inadequate for the task at hand. We develop a computational approach to appropriately utilize both data sources for estimating a regulatory network. The proposed approach is based on a three-step algorithm to estimate the underlying directed but cyclic network, that uses as input both perturbation screens and steady state gene expression data. In the first step, the algorithm determines causal orderings of the genes that are consistent with the perturbation data, by combining an exhaustive search method with a fast heuristic that in turn couples a Monte Carlo technique with a fast search algorithm. In the second step, for each obtained causal ordering, a regulatory network is estimated using a penalized likelihood based method, while in the third step a consensus network is constructed from the highest scored ones. Extensive computational experiments show that the algorithm performs well in reconstructing the underlying network and clearly outperforms competing approaches that rely only on a single data source. Further, it is established that the algorithm produces a consistent estimate of the regulatory network.\n",
      "    </p>\n",
      "   </abstract>\n",
      "   <funding-group>\n",
      "    <funding-statement>\n",
      "     The work was supported by National Institutes of Health grants 1RC1CA145444-01 to GM and 1R21GM101719-01A1 to GM and AS. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.\n",
      "    </funding-statement>\n",
      "   </funding-group>\n",
      "   <counts>\n",
      "    <page-count count=\"16\"/>\n",
      "   </counts>\n",
      "  </article-meta>\n",
      " </front>\n",
      " <body>\n",
      "  <sec id=\"s1\">\n",
      "   <title>\n",
      "    Introduction\n",
      "   </title>\n",
      "   <p>\n",
      "    The ability to reconstruct cellular networks plays an important role in our understanding of how genes interact with each other and the way information flows through them to regulate their expression levels. Such reconstructions heavily depend on the input data employed. The availability of data on the response of the cell to\n",
      "    <italic>\n",
      "     perturbations\n",
      "    </italic>\n",
      "    -either by knocking out or silencing genes- offers the possibility for improved network reconstructions and constitutes a key input in functional genomics. As pointed out in\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Markowetz1\">\n",
      "     [1]\n",
      "    </xref>\n",
      "    , high-dimensional phenotypic profiles obtained from perturbation experiments in the form of expression data offer the potential for obtaining a comprehensive view of cellular functions, even though they exhibit a number of limitations as outlined in\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Tresch1\">\n",
      "     [2]\n",
      "    </xref>\n",
      "    . A key problem is the fact that perturbation experiments only provide indirect information on gene interactions, as explained below\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Klamt1\">\n",
      "     [3]\n",
      "    </xref>\n",
      "    . Further, inferring large scale cellular networks from perturbation data is computationally challenging, and only a limited number of computational tools have been developed to address it. Some approaches are built on clustering of phenotypic profiles\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Piano1\">\n",
      "     [4]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Ohya1\">\n",
      "     [5]\n",
      "    </xref>\n",
      "    , which are based on the reasoning that functionally related genes should exhibit similar behavior under perturbations and hence cluster together. A tailor-made approach for the problem of estimating networks from perturbation data is the nested effects models (NEMs)\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Markowetz1\">\n",
      "     [1]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Tresch1\">\n",
      "     [2]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Markowetz2\">\n",
      "     [6]\n",
      "    </xref>\n",
      "    . NEMs are a special class of graphical models originally introduced to uncover the hierarchies among transcription factors based on observations of affected genes. More recently, NEMs have been extended to reconstruct regulatory networks by taking advantage of the nested structure of the observed perturbation effects, where for computational efficiency purposes, triplets of genes are used to assemble the global regulatory network. Extensions of this method that capture temporal effects by using perturbation time series measurements are described in\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Anchang1\">\n",
      "     [7]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Frhlich1\">\n",
      "     [8]\n",
      "    </xref>\n",
      "    . In response to the reconstruction problems presented in the DREAM challenges (see more information in the\n",
      "    <xref ref-type=\"sec\" rid=\"s3\">\n",
      "     Results\n",
      "    </xref>\n",
      "    section), methods like feed-forward loop down ranking (FFLDR)\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Pinna1\">\n",
      "     [9]\n",
      "    </xref>\n",
      "    and a\n",
      "    <inline-formula>\n",
      "     <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e001\" xlink:type=\"simple\"/>\n",
      "    </inline-formula>\n",
      "    -test based method coupled with ordinary differential equations to model temporal changes in expression data (Inferelator)\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Greenfield1\">\n",
      "     [10]\n",
      "    </xref>\n",
      "    were also developed.\n",
      "   </p>\n",
      "   <p>\n",
      "    The computational difficulty of reconstructing a network from data, alluded to above, stems from the fact that in order to capture the regulatory interactions one should consider all possible orderings of genes in the network (so that parent nodes influence child ones) and score the resulting network structures accordingly. The computational complexity of identifying all possible orderings of a set of nodes in a directed graph is exponential in the size of the graph. Hence, the approach based on nested effects models employs several heuristics for searching the space of orderings. Similarly,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Friedman1\">\n",
      "     [11]\n",
      "    </xref>\n",
      "    employs a Markov Chain Monte Carlo based search method and subsequent scoring of the resulting network structures.\n",
      "   </p>\n",
      "   <p>\n",
      "    Another set of approaches solely utilizes observational gene expression data that capture the system in steady state. A major technical tool for such reconstructions is graphical models\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Pearl1\">\n",
      "     [12]\n",
      "    </xref>\n",
      "    that encode a probability model over the genes through the underlying network. Over the last few years, a number of algorithms have been proposed in the literature for the estimation (reconstruction) of primarily Gaussian graphical models under the assumption of a sparse underlying structure (see\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "     [13]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Michailidis1\">\n",
      "     [14]\n",
      "    </xref>\n",
      "    for a discussion and references therein). The main shortcoming of these approaches is that graphical models for observational data are mostly capable of identifying dependencies between genes, rather than causal relations representing regulatory mechanisms. Further, the presence of more genes than available samples usually leads to very sparse reconstructions. It should be noted that recent work is geared towards identifying causal effects from observational data by employing the concept of intervention calculus\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Maathuis1\">\n",
      "     [15]\n",
      "    </xref>\n",
      "    . Also utilizing only steady state gene expression data is a method implemented in the PC-algorithm\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Spirtes1\">\n",
      "     [16]\n",
      "    </xref>\n",
      "    that starts from a complete undirected graph and recursively sparsifies it based on conditional independence decisions; directionality can only be inferred for a subset of edges (due to the issue of observational equivalence\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Pearl1\">\n",
      "     [12]\n",
      "    </xref>\n",
      "    ) and is added as a post-processing step.\n",
      "   </p>\n",
      "   <p>\n",
      "    Time course gene expression data have also been used to estimate gene regulatory networks, using two classes of models. In the first approach, called Granger causality, the predictive effect of genes on each other are used to estimate regulatory relationships among genes\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Yamaguchi1\">\n",
      "     [17]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-OpgenRhein1\">\n",
      "     [18]\n",
      "    </xref>\n",
      "    . In the second approach, known as dynamic Bayesian networks (DBNs), the framework of Bayesian networks is extended to incorporate biological networks with feedback loops\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Michailidis1\">\n",
      "     [14]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Ong1\">\n",
      "     [19]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Perrin1\">\n",
      "     [20]\n",
      "    </xref>\n",
      "    . More recently, penalization methods have been applied to improve the estimation of high-dimensional regulatory networks from small sample sizes\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Fujita1\">\n",
      "     [21]\n",
      "    </xref>\n",
      "    –\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie3\">\n",
      "     [24]\n",
      "    </xref>\n",
      "    .\n",
      "   </p>\n",
      "   <p>\n",
      "    The proposed computational approach in this study utilizes both gene expression data obtained from perturbation experiments, and an independent expression data set reflecting steady state behavior of the cell. The main steps are summarized in\n",
      "    <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "     Figure 1A\n",
      "    </xref>\n",
      "    . Specifically, based on the perturbation phenotyping data, we obtain a (large) set of causal orderings of the genes through a fast search algorithm which samples from the space of all possible orderings (see section Obtaining Causal Orderings from Perturbation Data for the definition of a causal ordering and algorithmic details). These orderings correspond to the inherent layering of nodes of the graph. Each set of orderings is then employed to obtain a directed acyclic regulatory graph/network (DAG) using the independent gene expression data set through an extension of a fast penalized likelihood method introduced in\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "     [13]\n",
      "    </xref>\n",
      "    . Further, the likelihood of every estimated graph is calculated. Finally, a\n",
      "    <italic>\n",
      "     consensus\n",
      "    </italic>\n",
      "    regulatory network (which can very well contain cycles when the true network is cyclic) is obtained by averaging a small set of the most likely DAGs obtained. The advantage of the proposed algorithm is that it utilizes both perturbation and steady state expression data, and once the set of gene orderings is determined, a theoretically rigorous and computationally fast likelihood-based method for estimating the underlying network structure is used. Further, causal orderings are determined by searching through the entire set of orderings consistent with the perturbation data, thus taking a global perspective, as opposed to competing methods (e.g. the nested effects model or the PC-algorithm) that only utilize information pertaining to direct neighbors of nodes and then assembling the network from such local estimates.\n",
      "   </p>\n",
      "   <fig id=\"pone-0082393-g001\" position=\"float\">\n",
      "    <object-id pub-id-type=\"doi\">\n",
      "     10.1371/journal.pone.0082393.g001\n",
      "    </object-id>\n",
      "    <label>\n",
      "     Figure 1\n",
      "    </label>\n",
      "    <caption>\n",
      "     <title>\n",
      "      Overview and details of the RIPE method.\n",
      "     </title>\n",
      "     <p>\n",
      "      (A) Overview of the RIPE algorithm. (B) First step to obtain a large set of causal orderings from a network with cycles. The network graph is decomposed into strongly connected components, so called super-nodes, (left) followed by a DFS on the strongly connected components (right). For example, the post-visit time of super-node A is 8, and thus A precedes all other nodes. The topological ordering of super-nodes is A\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e002\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      B\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e003\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      C\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e004\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      D. (C) Illustration of MC-DFS algorithm. Gene perturbation graph (left), DFS visit times for labeling #1 (middle), and DFS visit times for labeling #2 (right). (D) Depiction of a small network to illustrate the influence graph and the predictors used in the penalized likelihood estimation procedure. The true regulatory network (left), the influence graph under no noise (middle), observed influence graph, with false positive and false negative edges (right). Edges in the regulatory networks are shown in thick lines and additional edges in the influence graph are distinguished with narrow lines; red dash lines indicate false positives.\n",
      "     </p>\n",
      "    </caption>\n",
      "    <graphic mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.g001\" xlink:type=\"simple\"/>\n",
      "   </fig>\n",
      "   <p>\n",
      "    The proposed methodology has a number of additional advantages over competing methods. First, the underlying assumption in the estimation of regulatory networks from perturbation data is that the structure of the network does not change in different knockout experiments, and therefore observations from these experiments can be combined in order to estimate the underlying regulatory network. However, such an assumption is not fully valid: biological systems and regulatory networks are highly robust\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Stelling1\">\n",
      "     [25]\n",
      "    </xref>\n",
      "    ,\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Kitano1\">\n",
      "     [26]\n",
      "    </xref>\n",
      "    , which is believed to be the result of redundant regulatory mechanisms, and knocking out a single gene may trigger an alternate regulatory pathway, resulting in a different network structure. On the other hand, the main assumption of the methodology proposed in this paper is that the causal orderings of genes remain stable in different perturbation experiments, which is significantly less restrictive. Second, although the problem of reconstructing regulatory networks is computationally NP-hard, the computational complexity of the approximate algorithm proposed in this paper is considerably lower than most methods of network reconstructions directly from perturbation data. This is mainly due to the following facts: (i) the space of possible orderings is smaller than the space of graphs (even acyclic ones)\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Friedman1\">\n",
      "     [11]\n",
      "    </xref>\n",
      "    , and (ii) by employing the Monte Carlo sampling framework over the space of orderings, the approximate algorithm offers a tractable alternative in high dimensional settings. Our extensive experimental results indicate that the algorithm does not require an exhaustive search over the space of orderings, and a much smaller set of orderings often results in significant improvements over competing methods. Finally, it is known that perturbation data in the form of single gene knockouts do not provide sufficient information for estimation of regulatory networks and, in theory, all possible knockout combinations may be needed to fully discover a regulatory network. On the other hand, the indirect information in perturbation experiments is sufficient for estimating the causal orderings (see the subsection Obtaining Causal Orderings from Perturbation Data under\n",
      "    <xref ref-type=\"sec\" rid=\"s2\">\n",
      "     Methods\n",
      "    </xref>\n",
      "    ). Therefore, by breaking down the network reconstruction problem into three steps, not only can we achieve better computational complexity, but we can also transform a non-tractable problem into a sequence of tractable sub-problems. Further, the proposed approach offers the possibility to gain insight into the informational contributions of the two data sources used, and offers improved performance through systematic integration of the two sources of data.\n",
      "   </p>\n",
      "  </sec>\n",
      "  <sec id=\"s2\" sec-type=\"methods\">\n",
      "   <title>\n",
      "    Methods\n",
      "   </title>\n",
      "   <sec id=\"s2a\">\n",
      "    <title>\n",
      "     Method Overview\n",
      "    </title>\n",
      "    <p>\n",
      "     We introduce next a three-step algorithm, called\n",
      "     <italic>\n",
      "      RIPE\n",
      "     </italic>\n",
      "     (standing for Regulatory network Inference from joint Perturbation and Expression data), which incorporates both perturbation screens from knockout/knockdown experiments, as well as gene expression data usually reflecting steady state behavior of the cell, in order to address the problem of estimation of regulatory networks. We adopt the term\n",
      "     <italic>\n",
      "      steady-state\n",
      "     </italic>\n",
      "     data in our presentation for wild-type data. The data obtained from perturbation (knock-out/down) experiments are referred to as\n",
      "     <italic>\n",
      "      perturbation\n",
      "     </italic>\n",
      "     data. Note though that in practice, the actual “lab measurements” of the knockout effects are also considered in an equilibrium state. However, for presentation clarity, we differentiate the two above-mentioned sources of data. The main steps for RIPE are illustrated in the flowchart of\n",
      "     <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "      Figure 1A\n",
      "     </xref>\n",
      "     . In the first step, the data from the perturbation screens are employed to obtain a large collection of causal orderings. In the second step, each causal ordering is used in conjunction with the steady state data to obtain an estimated regulatory network through a penalized likelihood approach. Finally, in the third step a consensus network is constructed from the best networks obtained in the second step. A detailed description of these three steps follows.\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"s2b\">\n",
      "    <title>\n",
      "     Obtaining Causal Orderings from Perturbation Data\n",
      "    </title>\n",
      "    <sec id=\"s2b1\">\n",
      "     <title>\n",
      "      Estimating the Influence Matrix from Perturbation Data\n",
      "     </title>\n",
      "     <p>\n",
      "      Let\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e005\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      denote the\n",
      "      <italic>\n",
      "       binary influence matrix\n",
      "      </italic>\n",
      "      of size\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e006\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e007\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      representing the number of genes under study and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e008\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      the number of single genes that are knocked out/down. In general, the RIPE algorithm requires “perturbation” of individual genes. These perturbations can be in the form of knockouts and/or knockdowns as illustrated in the the section DREAM4 Challenge under\n",
      "      <xref ref-type=\"sec\" rid=\"s3\">\n",
      "       Results\n",
      "      </xref>\n",
      "      . Thus, all references to knockout experiments throughout the text also include single gene knockdowns. Each knockout experiment is repeated\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e009\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      times, and captures the effect of the knockout gene on the remaining genes. Also required are observations on the unperturbed network (repeated\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e010\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      times), to determine the baseline expression of the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e011\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      genes.\n",
      "     </p>\n",
      "     <p>\n",
      "      An entry in the influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e012\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      if the knockout experiment for gene\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e013\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is affecting gene\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e014\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , and 0 otherwise. To assess whether an entry is non-zero, meaning that gene\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e015\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is differentially expressed in the knockout experiment of gene\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e016\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      compared to its baseline expression, we can use e.g. a (moderated) 2-sample\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e017\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -test. Applying different cutoffs to the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e018\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -values obtained from such an analysis results in different number of non-zero entries in\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e019\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . In Section Determining the Influence Matrix under\n",
      "      <xref ref-type=\"sec\" rid=\"s3\">\n",
      "       Results\n",
      "      </xref>\n",
      "      , we describe a systematic procedure for determining the appropriate\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e020\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -value cutoff. It is therefore clear that as the quality of the perturbation experiments improves, and/or number of replicates for each perturbation experiments increases, fewer false positive/negative edges are present in the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e021\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . Well-conducted perturbation experiments with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e022\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      in the range of 2 to 5 replicates often provide the required level of accuracy.\n",
      "     </p>\n",
      "     <p>\n",
      "      From matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e023\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      one can obtain the\n",
      "      <italic>\n",
      "       directed influence graph\n",
      "      </italic>\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e024\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e025\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , where\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e026\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is the set of vertices and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e027\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      the set of the graph edges. The graph\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e028\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      contains an edge from node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e029\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      to node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e030\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      if the corresponding entry\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e031\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      .\n",
      "     </p>\n",
      "     <p>\n",
      "      It is worth noting that the RIPE algorithm can also be used in the case where\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e032\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , which is often of interest in the setting of regulatory networks. In this setting\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e033\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is the number of transcription factors (TFs), and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e034\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      the total number of genes under study. Assuming that there are no directed edges from target genes to TFs, causal orderings are then found only from TFs to other TFs as well as target genes. To simplify the exposition of statistical models, we discuss the details of the algorithm in the setting where\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e035\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , and defer the case of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e036\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      to Section Regulatory Network in Yeast, where we illustrate the application of RIPE in such a setting by estimating the gene regulatory network of yeast with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e037\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      perturbations on transcription factors, and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e038\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      total genes.\n",
      "     </p>\n",
      "    </sec>\n",
      "    <sec id=\"s2b2\">\n",
      "     <title>\n",
      "      Obtaining Causal Orderings based on the Influence Matrix\n",
      "     </title>\n",
      "     <p>\n",
      "      A critical step in the proposed approach is to obtain causal orderings of the genes from the influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e039\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . In case the influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e040\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      describes “acyclic” causal effects (i.e, the influence graph of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e041\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is acyclic),\n",
      "      <italic>\n",
      "       a single\n",
      "      </italic>\n",
      "      ordering of the underlying graph would suffice. However, most likely the perturbation matrix contains cyclic causal effects, due to the presence of feedback mechanisms in the regulatory network. Further, perturbation experiments usually yield noisy data, which could also result in cycles in the influence matrix, even if the underlying network is acyclic. Hence, one usually deals with an influence matrix whose underlying graph contains cycles and an individual causal ordering is not sufficient. We discuss next how one can obtain a\n",
      "      <italic>\n",
      "       set\n",
      "      </italic>\n",
      "      of causal orderings from the influence graph\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e042\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      .\n",
      "     </p>\n",
      "     <p>\n",
      "      We start by providing key definitions for a\n",
      "      <italic>\n",
      "       linear ordering\n",
      "      </italic>\n",
      "      of a set,\n",
      "      <italic>\n",
      "       topological ordering\n",
      "      </italic>\n",
      "      of an acyclic directed graph, and\n",
      "      <italic>\n",
      "       causal ordering\n",
      "      </italic>\n",
      "      of a directed graph. A\n",
      "      <italic>\n",
      "       linear ordering\n",
      "      </italic>\n",
      "      of the elements of a set is a collection of\n",
      "      <italic>\n",
      "       ordered pairs\n",
      "      </italic>\n",
      "      of the set elements, such that the ordering of each pair satisfies a certain criterion. In graph theory, an example of a linear ordering is a topological ordering. A\n",
      "      <italic>\n",
      "       topological ordering\n",
      "      </italic>\n",
      "      (known also as topological sort\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Dasgupta1\">\n",
      "       [27]\n",
      "      </xref>\n",
      "      ) of a directed acyclic graph\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e043\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      comprising of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e044\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      nodes, corresponds to a linear list of the form\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e045\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      of the nodes\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e046\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e047\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      denoting the label of the node in the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e048\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -th position in the ordered list. The ordering adheres to all partial relations\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e049\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      implied by the graph\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e050\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , where the relation\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e051\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is interpreted as “node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e052\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      precedes node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e053\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ”, i.e. there is an acyclic path from node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e054\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      to node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e055\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . We define the\n",
      "      <italic>\n",
      "       causal ordering\n",
      "      </italic>\n",
      "      of a directed graph to be the linear listing of nodes that corresponds to a valid\n",
      "      <italic>\n",
      "       depth-first search\n",
      "      </italic>\n",
      "      traversal of the graph, defined next. Before proceeding to the description of our algorithms, we want to emphasize a distinction between the terms\n",
      "      <italic>\n",
      "       topological sort\n",
      "      </italic>\n",
      "      and\n",
      "      <italic>\n",
      "       causal ordering\n",
      "      </italic>\n",
      "      . In acyclic graphs, a topological sorting is a special case of a causal ordering. In other words, the former refers to graphs with no cycles; the latter refers to linear orderings of causal effects induced by the influence graph, and are obtained from graphs that potentially have cycles. The difference is that in the latter ordering, not all partial relations of the form\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e056\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      hold.\n",
      "     </p>\n",
      "     <p>\n",
      "      A standard method in graph theory for obtaining a topological ordering of an acyclic graph is by employing the depth-first search (DFS) algorithm\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Dasgupta1\">\n",
      "       [27]\n",
      "      </xref>\n",
      "      ,\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Tarjan1\">\n",
      "       [28]\n",
      "      </xref>\n",
      "      which is outlined in\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g002\">\n",
      "       Figure 2\n",
      "      </xref>\n",
      "      . When the exact same algorithm is utilized in graphs with cycles, a causal ordering is obtained. DFS is a graph traversal algorithm. It “searches” the graph by traversing it “in-depth”. This means that when a node is discovered, the algorithm will continue searching for undiscovered nodes adjacent to the current one. When the algorithm reaches a “dead-end”, it backtracks until it finds previously visited nodes that have undiscovered neighboring nodes, and if this fails as well, it proceeds to new nodes not yet visited. The complete details of the algorithm are shown in\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g002\">\n",
      "       Figure 2\n",
      "      </xref>\n",
      "      , where a recursive implementation of the method is given (similar to the one in\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Dasgupta1\">\n",
      "       [27]\n",
      "      </xref>\n",
      "      ). Note that the algorithm saves the time of the first discovery of the node (pre-visit time) and the time of final departure of the node (post-visit time). Final departure for, say, node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e057\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      occurs when there are no paths starting from\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e058\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and leading to undiscovered nodes. The ordering is readily acquired by a descending sort of the post-visit times, as shown in the right panel of\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "       Figure 1B\n",
      "      </xref>\n",
      "      .\n",
      "     </p>\n",
      "     <fig id=\"pone-0082393-g002\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.g002\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Figure 2\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Algorithm for DFS.\n",
      "       </title>\n",
      "       <p>\n",
      "        Detailed steps of the algorithm for the DFS search.\n",
      "       </p>\n",
      "      </caption>\n",
      "      <graphic mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.g002\" xlink:type=\"simple\"/>\n",
      "     </fig>\n",
      "     <p>\n",
      "      For cyclic graphs, a mere application of DFS does not produce sufficient information to help us reconstruct the regulatory network. We would like to obtain as many causal orderings as possible, and evaluate each individual causal ordering using our penalized likelihood method described in the next section. To tackle the problem of obtaining a large set of causal orderings in graphs with cycles, we need to introduce the following two steps.\n",
      "     </p>\n",
      "     <list list-type=\"bullet\">\n",
      "      <list-item>\n",
      "       <p>\n",
      "        In step 1, we\n",
      "        <italic>\n",
      "         decompose\n",
      "        </italic>\n",
      "        the graph into its connected components. Specifically, a\n",
      "        <italic>\n",
      "         strongly connected component\n",
      "        </italic>\n",
      "        <xref ref-type=\"bibr\" rid=\"pone.0082393-Dasgupta1\">\n",
      "         [27]\n",
      "        </xref>\n",
      "        is a subgraph, such that there exists a path from every node to any other node in the subgraph. Hence, if we collapse each strongly connected component into a single super-node, the resulting graph is a DAG (see\n",
      "        <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "         Figure 1B\n",
      "        </xref>\n",
      "        , left panel). We then produce a\n",
      "        <italic>\n",
      "         topological sorting\n",
      "        </italic>\n",
      "        of the super-nodes. Note that since the super-nodes form a DAG, a mere topological sort is sufficient (see\n",
      "        <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "         Figure 1B\n",
      "        </xref>\n",
      "        , right panel).\n",
      "       </p>\n",
      "      </list-item>\n",
      "      <list-item>\n",
      "       <p>\n",
      "        In step 2, we produce a set of\n",
      "        <italic>\n",
      "         causal orderings\n",
      "        </italic>\n",
      "        for each super-node. Recall that in the presence of cycles, a causal ordering is not a topological ordering; instead, it is a linear listing of nodes that arises from a graph traversal using DFS. We propose two methods to obtain a set of causal orderings, each described in the sequel. The first involves an exhaustive DFS search and can be employed on strong components with relatively few nodes; the other couples DFS with Monte Carlo sampling and is suitable for large size strong components.\n",
      "       </p>\n",
      "      </list-item>\n",
      "     </list>\n",
      "     <p>\n",
      "      After completion of the above steps we combine the set of orderings corresponding to each super-node and thus we obtain the “universe” of causal orderings imposed by the influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e059\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . For the network graph depicted in\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "       Figure 1B\n",
      "      </xref>\n",
      "      , the universe of orderings includes\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e060\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e061\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e062\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e063\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e064\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e065\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      .\n",
      "     </p>\n",
      "     <p>\n",
      "      In the exhaustive DFS approach, we modify and extend Knuth's backtracking algorithm\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Knuth1\">\n",
      "       [29]\n",
      "      </xref>\n",
      "      that was proposed for generating all topological sorting arrangements of a DAG. The exhaustive search procedure is initiated at every node of the strongly connected component (thus, a parallel/concurrent implementation of this is feasible). This ensures that all possible causal orderings will be considered. Now, suppose that our search method has just discovered a new node, say node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e066\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . A key idea that ensures that all paths initiated from that node will be accounted, is to save all adjacent nodes of the newly visited node in a circular list. Suppose the list of node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e067\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      contains nodes\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e068\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . Then, DFS proceeds exactly as discussed above, i.e., it traverses the graph “in-depth”, starting from\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e069\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . After all paths originating from node\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e070\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      are visited, the algorithms backtracks to\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e071\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , and then the alternative paths will be explored by consulting our circular list (i.e., paths starting at\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e072\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , etc). However, exhaustively searching via backtracking has exponential complexity due to the huge number of combinations of paths that DFS can take. This makes the method practically infeasible for relatively large strongly connected components (e.g. larger than\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e073\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ). Nevertheless, it represents a useful tool for obtaining the universe of orderings in small size components.\n",
      "     </p>\n",
      "     <p>\n",
      "      For large size strong components, we develop a\n",
      "      <italic>\n",
      "       fast approximation algorithm\n",
      "      </italic>\n",
      "      , named\n",
      "      <italic>\n",
      "       MC-DFS\n",
      "      </italic>\n",
      "      , that incorporates ideas from Monte Carlo sampling techniques. MC-DFS consists of two simple steps: first, it employs a random labeling of the graph nodes; then, it runs DFS based on the current labeling. The workings of the MC-DFS heuristic algorithm can be best demonstrated – for\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e074\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      label permutations – with the example of\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "       Figure 1C\n",
      "      </xref>\n",
      "      . Suppose that the influence graph given by matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e075\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is the one depicted on the leftmost panel of the figure. The middle panel depicts labeling\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e076\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and the rightmost panel shows labeling\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e077\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . The DFS post- and pre-visit times for both cases are as shown in the figure. Given labeling\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e078\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , DFS produces the following causal ordering: G2\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e079\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G1\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e080\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G4\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e081\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G8\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e082\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G3\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e083\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G6\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e084\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G5\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e085\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G7. Given labeling\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e086\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , DFS produces: G4\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e087\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G8\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e088\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G3\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e089\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G6\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e090\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G2\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e091\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G1\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e092\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G5\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e093\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      G7. Thus, a large set of random permutations of the node labels, followed by an application of the DFS algorithm allows us to obtain a significant number of causal orderings and efficiently sample the space of all possible orderings. Obviously, the quality of the sampled space of orderings depends on the number of label permutations used (for some empirical assessment see the\n",
      "      <xref ref-type=\"sec\" rid=\"s3\">\n",
      "       Results\n",
      "      </xref>\n",
      "      and\n",
      "      <xref ref-type=\"sec\" rid=\"s4\">\n",
      "       Discussion\n",
      "      </xref>\n",
      "      sections). The complexity of DFS is\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e094\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      for a graph with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e095\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      nodes and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e096\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      edges. Thus, the total complexity of MC-DFS for a strong component of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e097\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      nodes and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e098\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      edges is\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e099\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , should one decide to generate\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e100\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      permutations.\n",
      "     </p>\n",
      "    </sec>\n",
      "   </sec>\n",
      "   <sec id=\"s2c\">\n",
      "    <title>\n",
      "     Estimation of Network Structure Using Gene Expression Data\n",
      "    </title>\n",
      "    <p>\n",
      "     As mentioned above, a gene regulatory network can be represented by a directed graph, whose adjacency matrix is denoted by\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e101\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . The element\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e102\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     if gene\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e103\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is directly regulated by gene\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e104\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e105\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     otherwise. In the setting of graphical models\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Pearl1\">\n",
      "      [12]\n",
      "     </xref>\n",
      "     , the nodes of the graph represent random variables\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e106\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and the edges capture associations between them.\n",
      "    </p>\n",
      "    <p>\n",
      "     It was shown in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "      [13]\n",
      "     </xref>\n",
      "     , that if the underlying network is a DAG and an ordering of its nodes is\n",
      "     <italic>\n",
      "      known\n",
      "     </italic>\n",
      "     , then estimating the network reduces to estimating its skeleton, a significantly simpler computational problem. Specifically, the causal effects of random variables in a DAG can be explained using\n",
      "     <italic>\n",
      "      structural equation models\n",
      "     </italic>\n",
      "     (SEM)\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Pearl1\">\n",
      "      [12]\n",
      "     </xref>\n",
      "     . In the setting where the data are normally distributed, SEM's can be represented based on linear functions explaining the relationship between each node and the set of its parents in the network:\n",
      "     <disp-formula id=\"pone.0082393.e107\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e107\" xlink:type=\"simple\"/>\n",
      "      <label>\n",
      "       (1)\n",
      "      </label>\n",
      "     </disp-formula>\n",
      "     Here,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e108\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     denotes the set of\n",
      "     <italic>\n",
      "      parents\n",
      "     </italic>\n",
      "     of node\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e109\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e110\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     's are latent variables representing the variation in each node unexplained by its parents (for normally distributed data,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e111\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     ). Finally, the coefficients\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e112\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     for the linear regression model represent the\n",
      "     <italic>\n",
      "      effect\n",
      "     </italic>\n",
      "     of gene\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e113\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     on\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e114\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     for\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e115\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     .\n",
      "    </p>\n",
      "    <p>\n",
      "     In the case of cyclic graphs, the above SEM representation for DAGs is not directly applicable. However, for each causal ordering from the previous section, the nodes of the graph can be reordered to obtain a DAG. Hence, using the above representation, the problem of estimating the structure of the DAGs corresponding to a causal ordering of nodes, say\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e116\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , can be posed as a penalized likelihood estimation problem as shown in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "      [13]\n",
      "     </xref>\n",
      "     . In particular, let\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e117\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     be the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e118\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     matrix of gene expression data, whose columns have been re-arranged according to the causal ordering\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e119\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and denote by\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e120\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     the submatrix obtained by columns of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e121\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     indexed by set\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e122\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . Then as shown in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "      [13]\n",
      "     </xref>\n",
      "     , the estimate of the adjacency matrix of DAGs under the general weighted lasso (or\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e123\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     ) penalty, is found by solving the following\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e124\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -regularized least squares problems for\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e125\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     <disp-formula id=\"pone.0082393.e126\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e126\" xlink:type=\"simple\"/>\n",
      "      <label>\n",
      "       (2)\n",
      "      </label>\n",
      "     </disp-formula>\n",
      "     where\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e127\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     denotes the first\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e128\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     elements of the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e129\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     th column of the adjacency matrix and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e130\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the tuning parameter for each lasso regression problem. In Section Choice of Parameters and Properties of the Algorithm, we discuss the choice of this tuning parameter. Finally,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e131\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     represents the weights of the lasso method; for the regular lasso penalty used here,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e132\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . In the RIPE algorithm, a natural extension of\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "      [13]\n",
      "     </xref>\n",
      "     is to use the above penalized likelihood estimation framework in order to estimate a DAG for each ordering\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e133\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , where\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e134\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the set of orderings found from the perturbation data. However, the perturbation data provide additional information regarding the influence of the genes in the network. In particular, in the absence of noise, the set of parents of each gene in the regulatory network are a subset of the set of parents in the influence graph. Using this observation, we generalize the penalized estimation problem in (2) to limit the set of variables in each penalized regression to those of the parents of node\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e135\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     in the influence graph, consistent with each ordering, which equates the set of all\n",
      "     <italic>\n",
      "      ancestors\n",
      "     </italic>\n",
      "     of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e136\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     in the regulatory network. In other words, for each ordering\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e137\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , the set of edges pointing to each gene in the regulatory graph is estimated by solving the following\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e138\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -regularized regression (lasso) problem:\n",
      "     <disp-formula id=\"pone.0082393.e139\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e139\" xlink:type=\"simple\"/>\n",
      "      <label>\n",
      "       (3)\n",
      "      </label>\n",
      "     </disp-formula>\n",
      "     where\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e140\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     denotes the set of parents of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e141\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     in the influence graph consistent with ordering\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e142\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e143\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e144\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     norm of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e145\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     .\n",
      "    </p>\n",
      "    <p>\n",
      "     To illustrate the optimization problem for estimation of DAGs in the second step of RIPE, consider the regulatory network in the left panel of\n",
      "     <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "      Figure 1D\n",
      "     </xref>\n",
      "     . The middle panel of the figure represents the ideal influence matrix, obtained when no errors are present in the perturbation data, and the right panel represents a realization of the influence graph with both false positive and false negative edges. In the first step of RIPE, causal orderings are determined based on the graph in the right panel of\n",
      "     <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "      Figure 1D\n",
      "     </xref>\n",
      "     . An example of such an ordering is\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e146\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . Note that in this case many orderings exist, due to the presence of cycles in the influence graph. In the second step of the RIPE algorithm, a penalized regression problem is solved for each node, where the set of predictors are the set of parents of the node in the right panel of\n",
      "     <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "      Figure 1D\n",
      "     </xref>\n",
      "     , consistent with the given ordering. In particular, the following regression problems are solved for\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e147\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     (here\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e148\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     denotes regression of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e149\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     on\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e150\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e151\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     ,\n",
      "     <italic>\n",
      "      ignoring\n",
      "     </italic>\n",
      "     for ease of presentation the corresponding penalty term):\n",
      "     <disp-formula id=\"pone.0082393.e152\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e152\" xlink:type=\"simple\"/>\n",
      "     </disp-formula>\n",
      "     Using the results of these regressions, the value of the penalized negative log-likelihood function is then determined for each of the estimated graphs. Based on these values, the “best” networks are then used to construct the consensus graph, as described next in the section discussing the third step of the RIPE algorithm.\n",
      "    </p>\n",
      "    <p>\n",
      "     Given the set of orderings\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e153\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     of cardinality\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e154\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , one needs to solve\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e155\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     separate penalized regression problems, and store their corresponding penalized negative log-likelihood values, where the computational cost of each of these problems is\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e156\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . However, this step is fully parallelizable and using\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e157\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     processors the complexity reduces to\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e158\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     .\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"s2d\">\n",
      "    <title>\n",
      "     Graph averaging: a consensus regulatory network\n",
      "    </title>\n",
      "    <p>\n",
      "     As mentioned above, the estimated influence matrix\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e159\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     from perturbation screens results in multiple orderings\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e160\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , either due to feedback regulatory mechanisms or noisy measurements. In such cases, the estimate of the adjacency matrix of the graph can be obtained from those corresponding to the set of orderings achieving the smallest penalized negative log-likelihood values. Therefore, the final step of the RIPE algorithm includes a model averaging procedure that combines the estimated DAGs from multiple orderings to construct a\n",
      "     <italic>\n",
      "      cyclic consensus network\n",
      "     </italic>\n",
      "     . This is illustrated with a small cyclic subnetwork in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s001\">\n",
      "      Figure S1\n",
      "     </xref>\n",
      "     . In this example, the true network includes a number of cycles, and the estimate from the RIPE algorithm correctly identifies some of these cycles (also see\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s012\">\n",
      "      Text S1\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "    <p>\n",
      "     Let\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e161\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     denote the lower\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e162\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     th quantile of the penalized negative log-likelihood values, denoted by\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e163\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e164\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     be the set of orderings with the lowest\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e165\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     penalized negative log-likelihood values. The RIPE estimate of the adjacency matrix is then defined as the consensus DAG:\n",
      "     <disp-formula id=\"pone.0082393.e166\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e166\" xlink:type=\"simple\"/>\n",
      "      <label>\n",
      "       (4)\n",
      "      </label>\n",
      "     </disp-formula>\n",
      "     Here,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e167\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the DAG estimate from the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e168\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -th ordering,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e169\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     denotes the\n",
      "     <italic>\n",
      "      confidence\n",
      "     </italic>\n",
      "     of each edge in the final estimate, and\n",
      "     <bold>\n",
      "      1\n",
      "     </bold>\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e170\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the indicator function which equals\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e171\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     when the condition\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e172\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     holds, and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e173\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     otherwise.\n",
      "    </p>\n",
      "    <p>\n",
      "     Consequently, the estimate of the edge set of the graph is defined as\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e174\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , where\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e175\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the threshold for including an edge. The value of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e176\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     determines the desired level of confidence, and can be chosen by the user depending on the application of interest. Nevertheless, the above formulation provides a flexible estimation framework, in which\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e177\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is considered a tuning parameter or can be set based on prior analyses (see the section Choice of Parameters and Properties of the Algorithm below and the\n",
      "     <xref ref-type=\"sec\" rid=\"s3\">\n",
      "      Results\n",
      "     </xref>\n",
      "     section for more details about the choice of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e178\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     ).\n",
      "    </p>\n",
      "    <p>\n",
      "     The RIPE algorithm also produces an estimate of the\n",
      "     <italic>\n",
      "      sign\n",
      "     </italic>\n",
      "     of each edge, as well as the\n",
      "     <italic>\n",
      "      magnitude\n",
      "     </italic>\n",
      "     of the effect, defined by the matrices\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e179\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e180\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     below:\n",
      "     <disp-formula id=\"pone.0082393.e181\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e181\" xlink:type=\"simple\"/>\n",
      "      <label>\n",
      "       (5)\n",
      "      </label>\n",
      "     </disp-formula>\n",
      "     where the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e182\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     .\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"s2e\">\n",
      "    <title>\n",
      "     Choice of Parameters and Properties of the Algorithm\n",
      "    </title>\n",
      "    <p>\n",
      "     Similarly to any other learning algorithm, the performance of the RIPE algorithm depends on the choices of tuning parameters. There are three tuning parameter that need to be determined: (i) the penalty coefficient\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e183\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , (ii) the likelihood quantile\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e184\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and (iii) the confidence threshold\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e185\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . Next, we discuss strategies for choosing these parameters.\n",
      "    </p>\n",
      "    <p>\n",
      "     In penalized regression settings,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e186\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     determines the weight of the penalty term in the optimization problem, with larger values of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e187\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     resulting in more sparse estimates. In\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "      [13]\n",
      "     </xref>\n",
      "     , the following error-based choice of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e188\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     was proposed for the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e189\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -th regression in (2),\n",
      "     <disp-formula id=\"pone.0082393.e190\">\n",
      "      <graphic position=\"anchor\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.e190\" xlink:type=\"simple\"/>\n",
      "      <label>\n",
      "       (6)\n",
      "      </label>\n",
      "     </disp-formula>\n",
      "     Here,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e191\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     is the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e192\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -quantile of the standard normal distribution and it can be shown that this choice of penalty controls the probability of falsely joining two unconnected ancestral components at level\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e193\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . Interestingly, numerical studies in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "      [13]\n",
      "     </xref>\n",
      "     show that the result of the analysis is not sensitive to the choice of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e194\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and values of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e195\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     result in comparable estimates.\n",
      "    </p>\n",
      "    <p>\n",
      "     Even though the choice of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e196\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     in\n",
      "     <xref ref-type=\"disp-formula\" rid=\"pone.0082393.e190\">\n",
      "      equation (6)\n",
      "     </xref>\n",
      "     controls the probability of false positives, it results in over-sparse estimates. Numerical studies in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie3\">\n",
      "      [24]\n",
      "     </xref>\n",
      "     strongly suggest the smaller value\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e197\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , which is used in the numerical studies in this paper.\n",
      "    </p>\n",
      "    <p>\n",
      "     Unlike the choice of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e198\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , our numerical studies indicate that the RIPE algorithm is not sensitive to the choices of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e199\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e200\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , and a wide range of values can be used for these parameters (see\n",
      "     <xref ref-type=\"sec\" rid=\"s3\">\n",
      "      Results\n",
      "     </xref>\n",
      "     and\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s002\">\n",
      "      Figures S2\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s003\">\n",
      "      S3\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s004\">\n",
      "      S4\n",
      "     </xref>\n",
      "     ). It is worth pointing out that since the value of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e201\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     determines the proportion of highest likelihoods used in constructing the consensus graph, when the perturbation data gives a reliable estimate of the influence matrix\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e202\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , multiple orderings from the first step of RIPE would result in true feedback cycles in regulatory networks, and therefore, the differences in corresponding likelihood values are mostly due to the inherent noise in expression data. As a result, in the ideal case with no errors in the influence graph, a value of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e203\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     would produce the “best” estimate. However, in practice, to avoid inferior estimates, outlier values of the likelihood should not be incorporated in the estimate. Our numerical analyses show that values of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e204\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     would result in comparable estimates.\n",
      "    </p>\n",
      "    <p>\n",
      "     Finally, the choice of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e205\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     determines the confidence of edges in the estimated consensus network: large values of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e206\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     result in edges that are more consistently present in all estimated graphs, while small values allow for less frequently present edges to be included in the final estimate. As with\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e207\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , our numerical studies indicate that large values of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e208\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     result in over-sparse estimates, and we recommend values of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e209\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     (see the\n",
      "     <xref ref-type=\"sec\" rid=\"s3\">\n",
      "      Results\n",
      "     </xref>\n",
      "     section and\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s002\">\n",
      "      Figures S2\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s003\">\n",
      "      S3\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s004\">\n",
      "      S4\n",
      "     </xref>\n",
      "     ). All estimates in the paper were obtained by setting\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e210\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e211\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , to achieve a balance between the standard performance measures of Precision and Recall.\n",
      "    </p>\n",
      "    <p>\n",
      "     To complete our discussion of the RIPE algorithm, it is worth noting that an accurate influence matrix\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e212\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     including binary information from single gene knockouts contains sufficient information for obtaining the causal orderings of the underlying regulatory network, but\n",
      "     <italic>\n",
      "      not\n",
      "     </italic>\n",
      "     its structure (see Lemma 1 in the Supporting Information (\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s012\">\n",
      "      Text S1\n",
      "     </xref>\n",
      "     ) for a formal statement and proof of this result). In the RIPE algorithm, abundantly available steady-state expression data are used to compensate for this lack of information. Further, we establish the asymptotic consistency of the network structure estimated using the penalized likelihood method (see Lemma 2 in the Supporting Information,\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s012\">\n",
      "      Text S1\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "   </sec>\n",
      "  </sec>\n",
      "  <sec id=\"s3\">\n",
      "   <title>\n",
      "    Results\n",
      "   </title>\n",
      "   <sec id=\"s3a\">\n",
      "    <title>\n",
      "     Preliminaries\n",
      "    </title>\n",
      "    <sec id=\"s3a1\">\n",
      "     <title>\n",
      "      Determining the Influence Matrix\n",
      "     </title>\n",
      "     <p>\n",
      "      In Section Estimating the Influence Matrix from Perturbation Data under\n",
      "      <xref ref-type=\"sec\" rid=\"s2\">\n",
      "       Methods\n",
      "      </xref>\n",
      "      we describe how to estimate an influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e213\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      from perturbation experiments using differential expression analysis. The analysis produces\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e214\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -values for each entry in the matrix, and by choosing a specific cutoff\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e215\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , we get an estimate of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e216\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ; note that a lower\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e217\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      gives rise to a sparser matrix. To select a reasonable cutoff, the number of edges in the influence graph for different p-values are plotted together with the size of the largest connected component, see\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g003\">\n",
      "       Figure 3\n",
      "      </xref>\n",
      "      . If the network is modularized, i.e. to some extent consists of nodes grouped together as illustrated in\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g001\">\n",
      "       Figure 1B\n",
      "      </xref>\n",
      "      , we would consider a drop in the size of the largest connected component indicative of a good choice of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e218\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -value. If only a few edges make the difference between a large and a small connected component, we have most likely found the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e219\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -value for which the “noise edges” have been removed to reveal the modularized structure. If such a drop in the size of the large component is missing, possible reasons can be that the underlying network does not exhibit a modularized structure, or that the signal strength of the experimental data is rather low to clearly reveal it. However, a plot of the type given in\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g003\">\n",
      "       Figure 3\n",
      "      </xref>\n",
      "      can still be useful in such a setting, since we can choose a\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e220\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -value based on the size of the connected component. An overly large connected component is not very realistic from a biological perspective, in addition to being computational demanding since it produces many potential orderings.\n",
      "     </p>\n",
      "     <fig id=\"pone-0082393-g003\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.g003\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Figure 3\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Influence graph characteristics versus\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e221\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        -value for DREAM4.\n",
      "       </title>\n",
      "       <p>\n",
      "        Number of edges (open triangles) in the influence graph and the size of the largest connected component (dots) versus cut-off\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e222\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        -value for differential expression. The data is based on five replications of the knockout and wild-type experiments for (A) 100-node network 1, (B) 100-node network 3, and (C) 100-node network 5 in the DREAM4 challenge.\n",
      "       </p>\n",
      "      </caption>\n",
      "      <graphic mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.g003\" xlink:type=\"simple\"/>\n",
      "     </fig>\n",
      "    </sec>\n",
      "    <sec id=\"s3a2\">\n",
      "     <title>\n",
      "      Performance Evaluation Criteria\n",
      "     </title>\n",
      "     <p>\n",
      "      Several evaluation criteria have been proposed for assessing the performance of different network reconstruction methods. However, the choice of criteria depends on the information available on the “true” regulatory network: in synthetic, or in silico examples, the gold standard is known, and Precision, Recall and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e223\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      measures are often used as standard metrics (with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e224\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      being the harmonic mean of the other two metrics). On the other hand, in real data applications, the gold standard is often unknown and the appropriate performance criteria should be determined based on the information at hand. In the section below, titled Regulatory Network in Yeast, the known protein-DNA interactions for yeast present in the BIOGRID database (release 3.1.74)\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Stark1\">\n",
      "       [30]\n",
      "      </xref>\n",
      "      are used as gold standards. To account for the fact that the BIOGRID database may not include all regulatory interactions, we carried out an experiment similar to the approach implemented in\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Maathuis1\">\n",
      "       [15]\n",
      "      </xref>\n",
      "      . More specifically, to assess the significance of the number of true positives for each of the estimates, 10,000 Erdös-Rényi random graphs with the same number of edges as each of the estimates were generated (using the R-package igraph\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Csardi1\">\n",
      "       [31]\n",
      "      </xref>\n",
      "      ), and the number of true positives in randomly generated graphs was used to approximate the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e225\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -value for significance of the number of true positives observed. The resulting\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e226\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      -value determines the likelihood of observing a given number of true positives in randomly generated graphs, and can be used directly for performance evaluation.\n",
      "     </p>\n",
      "    </sec>\n",
      "   </sec>\n",
      "   <sec id=\"s3b\">\n",
      "    <title>\n",
      "     DREAM4 Challenge\n",
      "    </title>\n",
      "    <p>\n",
      "     To illustrate the performance of the RIPE algorithm, we first evaluate it by using networks from the DREAM4 in-silico network challenge, which is part of a series of challenges\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Prill1\">\n",
      "      [32]\n",
      "     </xref>\n",
      "     , aiming at inferring gene regulatory networks from simulated data. The gold standard networks in DREAM4 were generated by extracting modules from two source networks: yeast and\n",
      "     <italic>\n",
      "      Escherichia coli\n",
      "     </italic>\n",
      "     . The modules were extracted so as to preserve the structural properties of the underlying network, such as degree distribution and network motifs (statistically overrepresented circuit elements)\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Marbach1\">\n",
      "      [33]\n",
      "     </xref>\n",
      "     . The underlying dynamics in the models include mRNA and protein dynamics simulated by stochastic differential equations with both experimental and internal noise added (an extension to the method in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Marbach2\">\n",
      "      [34]\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "    <p>\n",
      "     The simulated data sets available within DREAM4 consist of observations from the unperturbed network (\n",
      "     <italic>\n",
      "      wild-type\n",
      "     </italic>\n",
      "     ), perturbation experiments in which all genes are knocked-out one-by-one (\n",
      "     <italic>\n",
      "      knockouts\n",
      "     </italic>\n",
      "     ), perturbation experiments in which the activity of all genes are lowered one by one with a factor of two (\n",
      "     <italic>\n",
      "      knockdowns\n",
      "     </italic>\n",
      "     ), as well as small perturbation of all genes simultaneously (\n",
      "     <italic>\n",
      "      multifactorial\n",
      "     </italic>\n",
      "     ) and finally, time series.\n",
      "    </p>\n",
      "    <p>\n",
      "     Three of the five 100-node networks were selected for analysis; network 1 (\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     ), network 3 (\n",
      "     <italic>\n",
      "      Net3\n",
      "     </italic>\n",
      "     ) and network 5 (\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     ). These particular choices were made based on the differences in topologies, as well as on how well the network structures were predicted in terms of AUC (Area Under the ROC Curve) in the in-silico challenge. Specifically, several competing research teams submitted predictions of the network structures for the DREAM4 challenge.\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     was best predicted overall, while the structure of\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     was the most difficult to deduce. We chose to also analyze\n",
      "     <italic>\n",
      "      Net3\n",
      "     </italic>\n",
      "     since Networks 2–4 were predicted with comparable accuracy, with intermediate AUC values compared to\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     and\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     . The varying prediction accuracies can be explained by the differences exhibited by the various network structures; for example,\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     has a layered structure, with many nodes functioning as either parents or children in the graph, while\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     has a more complex topology with several short cycles and nodes with multiple parents.\n",
      "    </p>\n",
      "    <p>\n",
      "     The DREAM4 challenge only includes one replicate of each simulated experiment, and in order to assess the noise levels in the data, we simulated five replicates of each of the wild-type, knockdown, and knockout experiments, as well as one multifactorial data set for the networks of interest by using GeneNetWeaver 3.0. The DREAM4 default settings were used, excluding standardization of the simulated data.\n",
      "    </p>\n",
      "    <p>\n",
      "     Two methods tied for first place in predicting the topology for the 100-node networks in DREAM4; a\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e227\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -test based method\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Greenfield1\">\n",
      "      [10]\n",
      "     </xref>\n",
      "     and a method based on confidence matrices, pruned by down-ranking of feed-forward loops (FFLDR)\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Pinna1\">\n",
      "      [9]\n",
      "     </xref>\n",
      "     . As the two methods performed comparably, we chose to compare our method to FFLDR, for which the authors kindly provided their code. In\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Greenfield1\">\n",
      "      [10]\n",
      "     </xref>\n",
      "     , the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e228\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -test method is also combined with time-series data using ordinary differential equations to model the temporal changes in the gene expression (known as Inferelator). However, as this method was not submitted in the challenge of network topology prediction, and in addition utilizes time series data, we have not included it in the comparisons.\n",
      "    </p>\n",
      "    <p>\n",
      "     The influence matrix\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e229\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     was estimated by comparing the expression levels from perturbation experiments (five replicates for each knockout/knockdown) to the corresponding levels in the unperturbed network (five replicates of the wild-type data), as described in the Preliminaries section above (see\n",
      "     <xref ref-type=\"fig\" rid=\"pone-0082393-g003\">\n",
      "      Figure 3\n",
      "     </xref>\n",
      "     ). The\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e230\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -value 0.019 was selected for both\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     and\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     , while 0.003 was chosen for\n",
      "     <italic>\n",
      "      Net3\n",
      "     </italic>\n",
      "     . For the knockdown data, the same method was used to select the cutoff p-values as for the knockout data (see\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s005\">\n",
      "      Figure S5\n",
      "     </xref>\n",
      "     ). For FFLDR, the optimal performance for large-scale networks is obtained by directly comparing the expression levels in knockout experiments, and hence, the five replicates of the knockout data (not using the wild-type data for comparison) were averaged, and this signal was used as input to the method.\n",
      "    </p>\n",
      "    <p>\n",
      "     <xref ref-type=\"table\" rid=\"pone-0082393-t001\">\n",
      "      Table 1\n",
      "     </xref>\n",
      "     summarizes the performances of the PC-algorithm (PCALG, as implemented in the R-package pcalg), the Nested Effects Model (NEM, as implemented in the R-package nem), as well as the FFLDR and RIPE (based on 10,000 MC-DFS orderings) algorithms in reconstructing\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     ,\n",
      "     <italic>\n",
      "      Net3\n",
      "     </italic>\n",
      "     , and\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     . Since the DREAM4 challenge does not include suitable steady-state expression data, to employ PCALG and RIPE,\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e231\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     independent samples from a Gaussian distribution with mean\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e232\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and variance\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e233\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     were generated from each of the networks. The strengths of associations among genes were chosen as uniform random numbers in the range\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e234\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     and to handle the effect of cycles in the regulatory network, the magnitudes of strengths were then normalized according to the method described in\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie4\">\n",
      "      [35]\n",
      "     </xref>\n",
      "     .\n",
      "    </p>\n",
      "    <table-wrap id=\"pone-0082393-t001\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0082393.t001\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Table 1\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Reconstruction results for DREAM4 networks with simulated steady state data.\n",
      "      </title>\n",
      "     </caption>\n",
      "     <alternatives>\n",
      "      <graphic id=\"pone-0082393-t001-1\" mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.t001\" xlink:type=\"simple\"/>\n",
      "      <table>\n",
      "       <colgroup span=\"1\">\n",
      "        <col align=\"left\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "       </colgroup>\n",
      "       <thead>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          Method\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          Net1, KO\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net1\n",
      "          </italic>\n",
      "          , KD\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net3\n",
      "          </italic>\n",
      "          , KO\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net3\n",
      "          </italic>\n",
      "          , KD\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net5\n",
      "          </italic>\n",
      "          , KO\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net5\n",
      "          </italic>\n",
      "          , KD\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "        </tr>\n",
      "       </thead>\n",
      "       <tbody>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          ARACNE\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          17\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          41\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          24\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          17\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          41\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          24\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          10\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          48\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          16\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          10\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          48\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          16\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          16\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          51\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          25\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          16\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          51\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          25\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          PCALG\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          48\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          26\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          40\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          23\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          29\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          48\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          25\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          33\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          44\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          25\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          31\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          40\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          20\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          27\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          43\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          22\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          29\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          NEM\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          28\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          31\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          6\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          6\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          6\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          14\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          9\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          11\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          4\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          4\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          4\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          9\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          17\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          12\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          7\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          10\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          08\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          FFLDR\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          92\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          57\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          70\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          86\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          46\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          60\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          66\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          44\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          53\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          60\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          28\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          38\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          44\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          35\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          39\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          49\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          26\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          RIPE\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          80\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          71\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          75\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          74\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          46\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          56\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          65\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          49\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          56\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          61\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          30\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          40\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          59\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          47\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          52\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          56\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          42\n",
      "         </td>\n",
      "        </tr>\n",
      "       </tbody>\n",
      "      </table>\n",
      "     </alternatives>\n",
      "     <table-wrap-foot>\n",
      "      <fn id=\"nt101\">\n",
      "       <label/>\n",
      "       <p>\n",
      "        Performance measures, in percentages, for methods of reconstruction of DREAM4\n",
      "        <italic>\n",
      "         Net1\n",
      "        </italic>\n",
      "        ,\n",
      "        <italic>\n",
      "         Net3\n",
      "        </italic>\n",
      "        , and\n",
      "        <italic>\n",
      "         Net5\n",
      "        </italic>\n",
      "        , using both knockout (KO) and knockdown (KD) data. Steady state expression data is generated from structural equation models based on the true graph.\n",
      "       </p>\n",
      "      </fn>\n",
      "     </table-wrap-foot>\n",
      "    </table-wrap>\n",
      "    <p>\n",
      "     As a benchmark, we also evaluated the performance of the ARACNE procedure\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Margolin1\">\n",
      "      [36]\n",
      "     </xref>\n",
      "     on the simulated data. It is important to note that ARACNE estimates\n",
      "     <italic>\n",
      "      undirected\n",
      "     </italic>\n",
      "     networks. Thus, the estimates from ARACNE cannot be directly compared with the other methods mentioned above. Nonetheless, we applied ARACNE with a number of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e235\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -value cutoffs and found that the Bonferroni adjusted\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e236\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -value cutoff of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e237\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     offers the best performance compared to the true network.\n",
      "    </p>\n",
      "    <p>\n",
      "     It can be seen from\n",
      "     <xref ref-type=\"table\" rid=\"pone-0082393-t001\">\n",
      "      Table 1\n",
      "     </xref>\n",
      "     that RIPE outperforms the competing methods (see\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e238\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     measure), with the exception of knockdown data for\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     where FFLDR exhibits a slight edge. The differences in performance are more pronounced for the more complex\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     . As mentioned above,\n",
      "     <italic>\n",
      "      Net3\n",
      "     </italic>\n",
      "     and\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     pose more challenging reconstruction problems compared to\n",
      "     <italic>\n",
      "      Net1\n",
      "     </italic>\n",
      "     ; hence, all methods exhibit inferior performances in the reconstruction task. On the other hand, while knockout data represent ideal perturbation experiments, knockdown data correspond to less accurate ones. Therefore, the performances of NEM and FFLDR, that only employ perturbation screens, as well as RIPE, deteriorate in the case of knockdown data, whereas the performance of PCALG and ARACNE are not affected by the change in the perturbation data, as it uses as input only steady state expression data.\n",
      "    </p>\n",
      "    <p>\n",
      "     To further evaluate the performance of ARACNE, PCALG and RIPE, we also applied these methods to the multifactorial data. As described earlier, the multifactorial data set is obtained from non-i.i.d observations, which violate the underlying assumption of both PCALG and RIPE algorithms. In addition, this data set does not correspond to a steady-state setting. Interestingly, the results in\n",
      "     <xref ref-type=\"table\" rid=\"pone-0082393-t002\">\n",
      "      Table 2\n",
      "     </xref>\n",
      "     indicate that the performance of all three competitors deteriorates by roughly similar factors, which can be attributed to the lower quality of this multifactorial data set. On the other hand, the results show that even when the steady-state data violate the underlying assumptions of the model, the performance of the RIPE algorithm is comparable to that of FFLDR, particularly for the more complex structure of\n",
      "     <italic>\n",
      "      Net5\n",
      "     </italic>\n",
      "     .\n",
      "    </p>\n",
      "    <table-wrap id=\"pone-0082393-t002\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0082393.t002\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Table 2\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Reconstruction results for DREAM4 networks with multifactorial data.\n",
      "      </title>\n",
      "     </caption>\n",
      "     <alternatives>\n",
      "      <graphic id=\"pone-0082393-t002-2\" mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.t002\" xlink:type=\"simple\"/>\n",
      "      <table>\n",
      "       <colgroup span=\"1\">\n",
      "        <col align=\"left\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "        <col align=\"center\" span=\"1\"/>\n",
      "       </colgroup>\n",
      "       <thead>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          Method\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net1\n",
      "          </italic>\n",
      "          , KO\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net1\n",
      "          </italic>\n",
      "          , KD\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net3\n",
      "          </italic>\n",
      "          , KO\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net3\n",
      "          </italic>\n",
      "          , KD\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net5\n",
      "          </italic>\n",
      "          , KO\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "          <italic>\n",
      "           Net5\n",
      "          </italic>\n",
      "          , KD\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          P\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          R\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          F\n",
      "          <sub>\n",
      "           1\n",
      "          </sub>\n",
      "         </td>\n",
      "        </tr>\n",
      "       </thead>\n",
      "       <tbody>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          ARACNE\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          25\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          9\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          13\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          25\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          9\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          13\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          28\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          19\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          23\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          28\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          19\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          23\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          16\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          14\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          15\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          16\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          14\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          15\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          PCALG\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          24\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          09\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          13\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          24\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          09\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          13\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          39\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          15\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          22\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          39\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          15\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          22\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          21\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          08\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          12\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          21\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          08\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          12\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          NEM\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          28\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          31\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          6\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          6\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          6\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          14\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          9\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          11\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          4\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          4\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          4\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          9\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          17\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          12\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          7\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          10\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          8\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          FFLDR\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          92\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          57\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          70\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          86\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          46\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          60\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          66\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          44\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          53\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          60\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          28\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          38\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          44\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          35\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          39\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          49\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          26\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "        </tr>\n",
      "        <tr>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          RIPE\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          74\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          38\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          50\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          80\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          30\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          43\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          71\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          31\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          43\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          68\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          23\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          34\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          54\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          32\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          40\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          52\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          24\n",
      "         </td>\n",
      "         <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "          33\n",
      "         </td>\n",
      "        </tr>\n",
      "       </tbody>\n",
      "      </table>\n",
      "     </alternatives>\n",
      "     <table-wrap-foot>\n",
      "      <fn id=\"nt102\">\n",
      "       <label/>\n",
      "       <p>\n",
      "        Performance measures, in percentages, for methods of reconstruction of DREAM4\n",
      "        <italic>\n",
      "         Net1\n",
      "        </italic>\n",
      "        ,\n",
      "        <italic>\n",
      "         Net3\n",
      "        </italic>\n",
      "        , and\n",
      "        <italic>\n",
      "         Net5\n",
      "        </italic>\n",
      "        , using both knockout (KO) and knockdown (KD) data. Multifactorial data from DREAM4 challenge is used as steady-state expression data.\n",
      "       </p>\n",
      "      </fn>\n",
      "     </table-wrap-foot>\n",
      "    </table-wrap>\n",
      "    <p>\n",
      "     These results strongly indicate that combining perturbation screens with steady state expression data are beneficial to the regulatory network reconstruction problem, especially in settings where the perturbation data are rather noisy and the steady state data exhibit good quality. To better address this issue and obtain a deeper insight into the effect of noise on the perturbation data we undertake a number of experiments based on synthetic data.\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"s3c\">\n",
      "    <title>\n",
      "     Experiments with Synthetic Data\n",
      "    </title>\n",
      "    <p>\n",
      "     To assess the influence of the inputs and steps required by the various algorithms, we examine a number of settings both in small and large scale networks.\n",
      "    </p>\n",
      "    <sec id=\"s3c1\">\n",
      "     <title>\n",
      "      Small Directed Acyclic Graph\n",
      "     </title>\n",
      "     <p>\n",
      "      We start our discussion on synthetic data with the toy example illustrated in\n",
      "      <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s006\">\n",
      "       Figure S6\n",
      "      </xref>\n",
      "      . Specifically, we employ a randomly generated DAG of size\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e239\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      corresponding to the true regulatory network. To emulate possible regulatory mechanisms, the generated DAG includes a number of “hub” genes, as well as two genes that are not regulated by any other gene.\n",
      "     </p>\n",
      "     <p>\n",
      "      To obtain independent expression data (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e240\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ) consistent with the underlying DAG, an association weight of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e241\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      is assigned to all the edges in the graph and the available functions in the R-package pcalg are used to generate\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e242\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      independent samples of Gaussian random variables with mean 0 and variance 1, according to structural equation models that incorporate the influence of nodes of the graph on each other (see\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Kalisch1\">\n",
      "       [37]\n",
      "      </xref>\n",
      "      for more details).\n",
      "     </p>\n",
      "     <p>\n",
      "      The influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e243\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      corresponding to the perturbation data is generated as follows: given the adjacency matrix of a DAG\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e244\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , it is shown in\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie5\">\n",
      "       [38]\n",
      "      </xref>\n",
      "      that\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e245\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , where\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e246\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      denotes the identity matrix and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e247\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      the ceiling operator. We denote by\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e248\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      the ground truth influence matrix corresponding to the generated DAG. However, as previously discussed, in practice the influence matrix is extracted from gene expression data obtained from the perturbation experiments and hence is inherently noisy. Depicted in\n",
      "      <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s007\">\n",
      "       Figure S7\n",
      "      </xref>\n",
      "      is the matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e249\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      (left-most image) and three variants of that matrix that we examine; in the first one (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e250\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ) the direction of a small proportion of edges reversed (second image from the left) and in the second one (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e251\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ) edges are added (second image from the right); the third matrix (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e252\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ) includes both reversed edges, as well as an addition of extra edges (right-most image).\n",
      "     </p>\n",
      "     <p>\n",
      "      In this case, the small size of the network together with the relatively small amount of noise introduced in the influence matrix allows us to obtain\n",
      "      <italic>\n",
      "       all\n",
      "      </italic>\n",
      "      possible orderings; specifically,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e253\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      which contains the most number of orderings, has 12 strongly connected components of small size with a total of 3926 possible orderings, and hence can be easily handled with exhaustive search. On the other hand, the total number of orderings for\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e254\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e255\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e256\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      are 1, 1, and 2 respectively. However, we emphasize that obtaining the causal orderings of large strong components (i.e., larger than 15 nodes) using the exhaustive method is computationally expensive, and the MC-DFS heuristic becomes the only practical choice.\n",
      "     </p>\n",
      "     <p>\n",
      "      The results of applying the four methods under consideration to each of the input data sets are given in\n",
      "      <xref ref-type=\"table\" rid=\"pone-0082393-t003\">\n",
      "       Table 3\n",
      "      </xref>\n",
      "      . Note that NEM and FFLDR use only the four influence matrices\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e257\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      as input, while the PCALG only uses the steady state gene expression data\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e258\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ; finally, RIPE uses both of them. The numbers in parentheses correspond to the standard deviation of the metrics for PCALG and RIPE obtained from\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e259\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      replications of the steady state gene expression data\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e260\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . The numbers reported for RIPE, are obtained by considering all possible orderings generated using the exhaustive DFS algorithm for\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e261\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , and for\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e262\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      the consensus graph was obtained by setting\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e263\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e264\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . A graphical summary of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e265\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      measures over different versions of the influence matrix is given in\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g004\">\n",
      "       Figure 4A\n",
      "      </xref>\n",
      "      . It is worth noting that as discussed above, the choices of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e266\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and confidence threshold\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e267\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      do not critically affect the performance of RIPE and values of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e268\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e269\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      yield comparable results (see\n",
      "      <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s002\">\n",
      "       Figures S2\n",
      "      </xref>\n",
      "      ,\n",
      "      <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s003\">\n",
      "       S3\n",
      "      </xref>\n",
      "      ,\n",
      "      <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s004\">\n",
      "       S4\n",
      "      </xref>\n",
      "      ).\n",
      "     </p>\n",
      "     <p>\n",
      "      It can be seen that by combining perturbation screens and steady state gene expression data, RIPE clearly outperforms the FFLDR, NEM and PCALG methods that use a single source of data. Note that PCALG only uses the\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e270\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      data, thus the identical entries in the table. It is worth noting that although the performances of NEM, FFLDR and RIPE are affected by the increasing level of noise in the perturbation data (as expected), RIPE can compensate for this loss of accuracy by incorporating the additional information from the steady-state data. Obviously, all these effects would be magnified if a large number of spurious edges are added or a large number of true edges are deleted, thus introducing a significant amount of noise in the influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e271\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      . Interestingly, our results indicate that PCALG has a slight edge over NEM (both in case of DREAM data and the synthetic network). This may be due to the specific structure of the DAG under consideration (experiments show that NEM, which uses triplets of nodes to determine the order of the edges, often performs better in chain-type graphs, for example in DREAM4\n",
      "      <italic>\n",
      "       Net1\n",
      "      </italic>\n",
      "      ). The inferior performance of NEM in this setting can also be attributed to the fact that NEM, as originally proposed, performs well settings where the number of perturbation experiments is significantly smaller than the number of affected genes, which is not the case in our numerical experiments. On the other hand, RIPE takes a global view by constructing causal orderings, in addition to independently evaluating them with steady state gene expression data. Of particular interest is the significant deterioration in the performance of FFLDR in the cases of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e272\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e273\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      that indicates a vulnerability of the method in the presence of noise in its input data.\n",
      "     </p>\n",
      "     <fig id=\"pone-0082393-g004\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.g004\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Figure 4\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Performance of RIPE and competing methods on the reconstruction of synthetic networks.\n",
      "       </title>\n",
      "       <p>\n",
      "        (A) Average\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e274\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        measures for reconstruction using NEM, PCALG, FFLDR and RIPE on a network of size\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e275\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        .\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e276\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        corresponds to the ideal influence graph and\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e277\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        to\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e278\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        represent increasing levels of noise in perturbation data (see also\n",
      "        <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s007\">\n",
      "         Figure S7\n",
      "        </xref>\n",
      "        ). (B)–(C) Average\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e279\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        measures for reconstruction of synthetic regulatory networks using PCALG, FFLDR and RIPE for different levels of false positive and negative noise in perturbation data. Numbers in parentheses indicate the expected number of false edges in each case. The true graph is an acyclic graph (DAG) of size\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e280\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        in (B) and a cyclic graph of size\n",
      "        <inline-formula>\n",
      "         <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e281\" xlink:type=\"simple\"/>\n",
      "        </inline-formula>\n",
      "        in (C).\n",
      "       </p>\n",
      "      </caption>\n",
      "      <graphic mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.g004\" xlink:type=\"simple\"/>\n",
      "     </fig>\n",
      "     <table-wrap id=\"pone-0082393-t003\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.t003\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Table 3\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Reconstruction results for synthetic networks subject to error in the perturbation data.\n",
      "       </title>\n",
      "      </caption>\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone-0082393-t003-3\" mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.t003\" xlink:type=\"simple\"/>\n",
      "       <table>\n",
      "        <colgroup span=\"1\">\n",
      "         <col align=\"left\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "        </colgroup>\n",
      "        <thead>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           Method\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <inline-formula>\n",
      "            <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e282\" xlink:type=\"simple\"/>\n",
      "           </inline-formula>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <inline-formula>\n",
      "            <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e283\" xlink:type=\"simple\"/>\n",
      "           </inline-formula>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <inline-formula>\n",
      "            <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e284\" xlink:type=\"simple\"/>\n",
      "           </inline-formula>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <inline-formula>\n",
      "            <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e285\" xlink:type=\"simple\"/>\n",
      "           </inline-formula>\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           P\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           R\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           P\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           R\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           P\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           R\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           P\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           R\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "         </tr>\n",
      "        </thead>\n",
      "        <tbody>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           54(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           52(10)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           53(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           54(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           54(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           52(10)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           53(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           52(10)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           53(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           54(8)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           52(10)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           53(8)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           NEM\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           50\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           33\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           40\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           55\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           33\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           39\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           36\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           33\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           41\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           21\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           28\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           24\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           100\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           67\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           79\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           40\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           92\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           67\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           77\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           33\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           36\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           33\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           28\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           30\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           95(4)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           99(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           97(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           79(4)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           90(4)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           94(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           92(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           100(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61(5)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           84(6)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           71(5)\n",
      "          </td>\n",
      "         </tr>\n",
      "        </tbody>\n",
      "       </table>\n",
      "      </alternatives>\n",
      "      <table-wrap-foot>\n",
      "       <fn id=\"nt103\">\n",
      "        <label/>\n",
      "        <p>\n",
      "         Average performances measures, in percentages, for methods of network reconstruction in the synthetic networks. Numbers in parentheses show standard deviations for methods based on simulated steady state data (PCALG and RIPE).\n",
      "         <inline-formula>\n",
      "          <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e286\" xlink:type=\"simple\"/>\n",
      "         </inline-formula>\n",
      "         indicates the ideal influence graph and\n",
      "         <inline-formula>\n",
      "          <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e287\" xlink:type=\"simple\"/>\n",
      "         </inline-formula>\n",
      "         to\n",
      "         <inline-formula>\n",
      "          <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e288\" xlink:type=\"simple\"/>\n",
      "         </inline-formula>\n",
      "         represent noisy versions of the influence graph (see\n",
      "         <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s007\">\n",
      "          Figure S7\n",
      "         </xref>\n",
      "         ).\n",
      "        </p>\n",
      "       </fn>\n",
      "      </table-wrap-foot>\n",
      "     </table-wrap>\n",
      "     <p>\n",
      "      Finally, to assess the effect of approximation used in MC-DFS, in comparison to having the universe of orderings, we estimated the regulatory network from\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e289\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      with different number of orderings. The average values of the Precision, Recall and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e290\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      measures over 100 replications for different number of orderings considered are given in\n",
      "      <xref ref-type=\"table\" rid=\"pone-0082393-t004\">\n",
      "       Table 4\n",
      "      </xref>\n",
      "      . All estimates were obtained by using\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e291\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e292\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      for estimating the consensus graph. It can be seen that, in comparison to the estimate obtained by evaluating all 3926 ordering using exhaustive DFS, the one obtained by considering a random subset of a small number of orderings provides comparable results. In particular, aside from a slight increase in recall, as the number of orderings increases, the performance of the method with only 200 orderings obtained using MC-DFS is comparable to utilizing the universe of orderings generated from an exhaustive search with DFS. This result suggests that the MC-DFS algorithm represents a viable alternative for settings involving a large size strongly connected component, and as a result, the RIPE algorithm is not highly sensitive to the number of orderings used to reconstruct the network.\n",
      "     </p>\n",
      "     <table-wrap id=\"pone-0082393-t004\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.t004\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Table 4\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Impact of increasing number of orderings used in the RIPE algorithm.\n",
      "       </title>\n",
      "      </caption>\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone-0082393-t004-4\" mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.t004\" xlink:type=\"simple\"/>\n",
      "       <table>\n",
      "        <colgroup span=\"1\">\n",
      "         <col align=\"left\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "        </colgroup>\n",
      "        <thead>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           Number of Orderings\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           P\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           R\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "         </tr>\n",
      "        </thead>\n",
      "        <tbody>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           100\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           60\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           81\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           69\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           200\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           82\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           1000\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           82\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           2000\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           60\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           3926 (all)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           84\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           71\n",
      "          </td>\n",
      "         </tr>\n",
      "        </tbody>\n",
      "       </table>\n",
      "      </alternatives>\n",
      "      <table-wrap-foot>\n",
      "       <fn id=\"nt104\">\n",
      "        <label/>\n",
      "        <p>\n",
      "         Average performance measures, in percentages, for RIPE in the synthetic network\n",
      "         <inline-formula>\n",
      "          <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e293\" xlink:type=\"simple\"/>\n",
      "         </inline-formula>\n",
      "         .\n",
      "        </p>\n",
      "       </fn>\n",
      "      </table-wrap-foot>\n",
      "     </table-wrap>\n",
      "    </sec>\n",
      "    <sec id=\"s3c2\">\n",
      "     <title>\n",
      "      Effect of False Positive and Negative Errors in Perturbation Data\n",
      "     </title>\n",
      "     <p>\n",
      "      To analyze the effect of false positive and negative errors in the perturbation data on the performance of the RIPE algorithm, a randomly generated DAG of size\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e294\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      was used as the true regulatory network. The influence matrix\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e295\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      corresponding to the true perturbation data, and independent expression data of size\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e296\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      consistent with the underlying DAG were generated according to the methods explained in the previous section.\n",
      "     </p>\n",
      "     <p>\n",
      "      To emulate the effect of false positive and negative errors, three different scenarios were considered: false positive errors (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e297\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ), in the form of edges not present in the true influence graph, false negative edges (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e298\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ), in the form of missing edges compared to the true graph, and both false positive and false negative edges (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e299\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ). Additionally, to assess the effect of increasing noise levels, 3 levels of noise in each of the above scenarios were considered. To determine the appropriate noise levels, the number of edges in\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e300\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      (198) were used as calibration, and the proportion of false positive and false negative edges were adjusted so that each randomly perturbed influence matrix included the same number of expected false edges. The results were 9 noisy influence graphs under each of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e301\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e302\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e303\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      settings with roughly 25, 50 and 75 false edges.\n",
      "     </p>\n",
      "     <p>\n",
      "      Considering the inefficiency of NEM for network estimation in high dimensional settings especially when the number of effect genes is small, we focus on the performance of FFLDR and RIPE, using PCALG as a benchmark. The results for RIPE were obtained based on 1000 randomly generated orderings for the cases where the influence graphs were generated with false positives (since the true graph is a DAG, the case of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e304\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      amounts to a single ordering).\n",
      "     </p>\n",
      "     <p>\n",
      "      The performance of the above methods in each of the input data sets are given in\n",
      "      <xref ref-type=\"table\" rid=\"pone-0082393-t005\">\n",
      "       Table 5\n",
      "      </xref>\n",
      "      , where the results for RIPE and PCALG correspond to averages over\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e305\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      independent draws of the steady state data.\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g004\">\n",
      "       Figure 4B\n",
      "      </xref>\n",
      "      summarizes the values of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e306\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      measures for different methods. These results reveal a number of interesting aspects of FFLDR and RIPE algorithms. First, as expected, increasing levels of false negatives impact the performance of RIPE, while FFLDR could be severely impacted by high false positive rates. Secondly, while the worst-case performance of RIPE (for the case of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e307\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ) matches that of FFLDR, the proposed data integration framework can result in significant gain in network estimation accuracy in other scenarios. While high quality perturbation screens greatly improve the performance of both, by combining steady state data and perturbation screens, the RIPE algorithm could compensate for the inaccuracy of the perturbation data, whereas estimation based on perturbation data alone can result in FFLDR estimates that are inferior to those of PCALG. Finally,\n",
      "      <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s008\">\n",
      "       Figure S8\n",
      "      </xref>\n",
      "      shows the improvement with increasing number of orderings in\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e308\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e309\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e310\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      for inference using the influence graph with highest level of false positives (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e311\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ). It can be seen that although the three measures moderately improve with higher number of orderings, a small number of MC-DFS orderings are sufficient for acceptable performance of the RIPE algorithm. Note that due to the acyclicity of the underlying graph, different levels of false negative errors correspond to a single ordering for the influence graph, and hence a similar comparison for the case of false negatives is not relevant.\n",
      "     </p>\n",
      "     <table-wrap id=\"pone-0082393-t005\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.t005\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Table 5\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Impact of increasing false positive and negative errors in perturbation data on estimation of acyclic graphs.\n",
      "       </title>\n",
      "      </caption>\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone-0082393-t005-5\" mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.t005\" xlink:type=\"simple\"/>\n",
      "       <table>\n",
      "        <colgroup span=\"1\">\n",
      "         <col align=\"left\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "        </colgroup>\n",
      "        <thead>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            P\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            R\n",
      "           </italic>\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "         </tr>\n",
      "        </thead>\n",
      "        <tbody>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           NO ERR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           99\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           97(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           100\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           97(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           98\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           97(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            FP\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.50%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           96(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           65\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           95(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           84\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           97(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           1%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           48\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           94(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           40\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           91(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           62\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           96(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           1.50%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           45\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           30\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           77(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           97\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           91(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            FN\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           10%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           90\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           90(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           93\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           92(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           87(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           20%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           85\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           85(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           86\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           85\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           30%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           74(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           76\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           80(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           71\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           69(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            FP\n",
      "           </italic>\n",
      "           +\n",
      "           <italic>\n",
      "            FN\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.25%, 5%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           74\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           92(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           65\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           92(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           86\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           93(2)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.5%, 10%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           58\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           52\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           66\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           87(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.75%, 15%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           70(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           56\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           48\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           89(2)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           68\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           87(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "        </tbody>\n",
      "       </table>\n",
      "      </alternatives>\n",
      "      <table-wrap-foot>\n",
      "       <fn id=\"nt105\">\n",
      "        <label/>\n",
      "        <p>\n",
      "         Average performances measures, in percentages for PCALG, FFLDR and RIPE in the synthetic DAG of size\n",
      "         <italic>\n",
      "          p\n",
      "         </italic>\n",
      "         = 100 with different error structures. Numbers in parentheses indicate the standard deviation of the estimates over 50 draws of simulated data (only for PCALG and RIPE).\n",
      "        </p>\n",
      "       </fn>\n",
      "      </table-wrap-foot>\n",
      "     </table-wrap>\n",
      "    </sec>\n",
      "    <sec id=\"s3c3\">\n",
      "     <title>\n",
      "      Large Cyclic Graph (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e312\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      )\n",
      "     </title>\n",
      "     <p>\n",
      "      Our final numerical experiment with synthetic data compares the performance of RIPE with those of PCALG and FFLDR in reconstructing large cyclic graphs in the presence of both false positive and negative noise in the perturbation data. The setting of this simulation is similar to that of the previous section, with the main difference being the size of the graph and presence of cycles (feedback loops) in the true graph. In particular, a random cyclic graph with\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e313\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      nodes and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e314\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      edges was generated, and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e315\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      samples were generated from zero mean, unit variance Gaussian random variables, as steady state expression levels from the true network.\n",
      "     </p>\n",
      "     <p>\n",
      "      As before, three different scenarios were considered: false positive errors (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e316\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ), false negative edges (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e317\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ), and both false positive and false negative edges (\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e318\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ) at 3 noise levels in each of the above scenarios. The noise levels were set up so that approximately 200, 400, and 600 erroneous edges were included in each of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e319\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      ,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e320\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      , and\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e321\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      settings.\n",
      "     </p>\n",
      "     <p>\n",
      "      The performance of the above methods for each of the input settings is given in\n",
      "      <xref ref-type=\"table\" rid=\"pone-0082393-t006\">\n",
      "       Table 6\n",
      "      </xref>\n",
      "      , where the results for RIPE (with 1000 random orderings) and PCALG are averages over\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e322\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      independent drawings of the steady state data.\n",
      "      <xref ref-type=\"fig\" rid=\"pone-0082393-g004\">\n",
      "       Figure 4C\n",
      "      </xref>\n",
      "      summarizes the values of\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e323\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      measures for different methods. These results confirm the findings of the previous section. In particular, RIPE outperforms the other two algorithms in all the simulated settings, and the difference between the performances of RIPE and FFLDR is magnified as more false positive edges are added to the perturbation graph. Finally, as expected, the PCALG does not compare favorably with the other two methods in the setting of cyclic graphs.\n",
      "     </p>\n",
      "     <table-wrap id=\"pone-0082393-t006\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0082393.t006\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Table 6\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Impact of increasing false positive and negative errors in perturbation data on estimation of high dimensional cyclic graphs.\n",
      "       </title>\n",
      "      </caption>\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone-0082393-t006-6\" mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.t006\" xlink:type=\"simple\"/>\n",
      "       <table>\n",
      "        <colgroup span=\"1\">\n",
      "         <col align=\"left\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "         <col align=\"center\" span=\"1\"/>\n",
      "        </colgroup>\n",
      "        <thead>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            F\n",
      "           </italic>\n",
      "           <sub>\n",
      "            1\n",
      "           </sub>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            P\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"3\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            R\n",
      "           </italic>\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           PCALG\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           FFLDR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           RIPE\n",
      "          </td>\n",
      "         </tr>\n",
      "        </thead>\n",
      "        <tbody>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           NO ERR\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           58\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           63(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           89\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           43\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           50(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            FP\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.02%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           47\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           63(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           63\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           38\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           51(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.04%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           39\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           43\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           80(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           35\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           50(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.06%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           34\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           59(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           33\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           35\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           50(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            FN\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           2%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           57\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           88\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           43\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           49(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           4%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           55\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           59(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           82\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           81(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           41\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           47(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           6%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           54\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           59(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           82\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           81(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           40\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           46(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           <italic>\n",
      "            FP\n",
      "           </italic>\n",
      "           +\n",
      "           <italic>\n",
      "            FN\n",
      "           </italic>\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.01%, 1%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           54\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           62(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           73\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           83(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           43\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           49(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.02%, 2%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           48\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           64\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           80(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           39\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           50(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "         <tr>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\"/>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           0.03%, 3%\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           27(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           42\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           61(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           31(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           51\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           79(1)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           25(3)\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           35\n",
      "          </td>\n",
      "          <td align=\"left\" colspan=\"1\" rowspan=\"1\">\n",
      "           50(1)\n",
      "          </td>\n",
      "         </tr>\n",
      "        </tbody>\n",
      "       </table>\n",
      "      </alternatives>\n",
      "      <table-wrap-foot>\n",
      "       <fn id=\"nt106\">\n",
      "        <label/>\n",
      "        <p>\n",
      "         Average performances measures, in percentages, for PCALG, FFLDR and RIPE in the synthetic cyclic network of size\n",
      "         <italic>\n",
      "          p\n",
      "         </italic>\n",
      "         = 1000 with different error structures. Numbers in parentheses indicate the standard deviation of the estimates over 5 draws of simulated data (only for PCALG and RIPE).\n",
      "        </p>\n",
      "       </fn>\n",
      "      </table-wrap-foot>\n",
      "     </table-wrap>\n",
      "    </sec>\n",
      "   </sec>\n",
      "   <sec id=\"s3d\">\n",
      "    <title>\n",
      "     Regulatory Network in Yeast\n",
      "    </title>\n",
      "    <p>\n",
      "     To evaluate our method on real data, we explore the transcription factor (TF) regulatory network in\n",
      "     <italic>\n",
      "      Saccharomyces cerevisiae\n",
      "     </italic>\n",
      "     . The influence matrix\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e324\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     was estimated from a large transcription factor knockout experiment\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Hu1\">\n",
      "      [39]\n",
      "     </xref>\n",
      "     , that was subsequently reanalyzed by\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Reimand1\">\n",
      "      [40]\n",
      "     </xref>\n",
      "     . Briefly, the experiment consists of 269 knockouts or knockdowns of yeast transcription factors, investigated by hybridization to microarrays under normal conditions. In total, 588 custom-made two-color microarrays with wild-type standard total RNA used as reference strain were employed in four batches and three different strains. Reanalysis of the data involved within-array normalization, background correction, between array-normalization and corrections for batch effects and strain effects. The genes were ranked using a moderated\n",
      "     <italic>\n",
      "      t\n",
      "     </italic>\n",
      "     -statistic, and FDR-correction was applied based on the method of\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-Benjamini1\">\n",
      "      [41]\n",
      "     </xref>\n",
      "     .\n",
      "    </p>\n",
      "    <p>\n",
      "     We extracted the expression values for the genes corresponding to the 269 TF knockouts (resulting in a square matrix of expression values) and a\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e325\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -value cut-off was chosen to 0.002 based on the same type of plot as for the DREAM4 networks (see\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s009\">\n",
      "      Figure S9\n",
      "     </xref>\n",
      "     ). The resulting influence matrix corresponds to a graph with a strongly connected component of size 113.\n",
      "    </p>\n",
      "    <p>\n",
      "     The steady state experimental data employed come from a publicly available data set (ArrayExpress E-TABM-773) and has been used (e.g., see\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0082393-vanWageningen1\">\n",
      "      [42]\n",
      "     </xref>\n",
      "     ) to assess the day-to-day variation in large yeast array experiments. The data contains 200 samples, with samples from each day hybridized against a pool of wild-type strains. Although the samples are not necessarily independent, or identically distributed (due to batch effects, temporal correlations etc), here we use this dataset as an approximation for i.i.d measurements. For the strong components whose size prohibits an exhaustive search, we apply the MC-DFS heuristic with\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e326\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     random permutations.\n",
      "    </p>\n",
      "    <p>\n",
      "     The performance of the competing methods is based on the strategy outlined in the Preliminaries section above.\n",
      "     <xref ref-type=\"fig\" rid=\"pone-0082393-g005\">\n",
      "      Figure 5\n",
      "     </xref>\n",
      "     shows the histogram of the number of true positives for randomly generated graphs of the same size as each of the reconstructed networks, as well as those obtained from three different methods Note that the NEM algorithm failed to produce an estimate of the network in this case after 10 days of run time and thus its results were not included. Interestingly, the RIPE estimate is the only method for which the number of true positives are significantly larger (at 5% significance level) than that obtained in a random graph (0.0185 for RIPE compared to 0.1265 for FFLDR and 0.5543 for PCALG). These findings further highlight the advantage of the proposed approach over existing methods.\n",
      "    </p>\n",
      "    <fig id=\"pone-0082393-g005\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0082393.g005\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Figure 5\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Performance evaluation for the reconstruction of the yeast regulatory network.\n",
      "      </title>\n",
      "      <p>\n",
      "       Number of true positives for each method, in comparison to the BIOGRID database, as well as a histogram for the number of true positives in randomly generated networks of the same size are shown. The p-values are obtained based on 10,000 randomly generated networks.\n",
      "      </p>\n",
      "     </caption>\n",
      "     <graphic mimetype=\"image\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.g005\" xlink:type=\"simple\"/>\n",
      "    </fig>\n",
      "    <p>\n",
      "     A comparison of the estimated graphs using these three methods indicates a significant overlap between RIPE and FFLDR reconstructions: specifically, out of 520 edges detected in RIPE and 622 edges in FFLDR, 225 edges are in common. On the other hand, the PCALG reconstruction has considerable less overlap with the other two estimates: 8 and 12 common edges with RIPE and FFLDR, respectively. The significant overlap between the FFLDR and RIPE estimates suggests that some of the edges detected by both methods may indeed correspond to true regulatory interactions in yeast, which are not included in the BIOGRID database. Such results could be used as a starting point for designing the corresponding validation experiments.\n",
      "    </p>\n",
      "    <p>\n",
      "     As with the DREAM4 data, we also applied ARACNE for estimation of the yeast network, and found that the Bonferroni adjusted p-value cutoff of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e327\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     results in the best estimate, with 131 true positives (compared to BIOGRID) and 5594 total edges. This indicates that the estimated network based ARACNE is significantly denser compared to all other estimators. Evaluation of the significance of the number of true positives using the random graph method described above gives a\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e328\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -value of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e329\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     , which indicates that ARACNE has a significantly larger proportion of true positives compared to FFLDR and PCALG. This is somewhat in agreement with the analysis on the DREAM4 data, where the combined performance of ARACNE based on the\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e330\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     measure was affected by its low Recall rate compared to the other methods.\n",
      "    </p>\n",
      "    <p>\n",
      "     As pointed out in the\n",
      "     <xref ref-type=\"sec\" rid=\"s2\">\n",
      "      Methods\n",
      "     </xref>\n",
      "     section, the RIPE algorithm can also be applied in the case where\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e331\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     . To illustrate this, we next describe the application of the RIPE algorithm for estimation of the entire regulatory network of yeast with\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e332\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     TF's and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e333\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     total genes. In such settings, one often obtains perturbation data on a subset of genes of interest (here the set of TFs) and is interested in obtaining an estimate of the\n",
      "     <italic>\n",
      "      regulatory interactions among the set of perturbed genes and all other genes\n",
      "     </italic>\n",
      "     in the network. In this example, this amounts to a network with edges from each of the 269 TF's to each of the genes (TF's and TG's) in the network. Specifically, this corresponds to a 2-layer graph consisting of edges amongst TF's, as well as edges between TF's and TG's.\n",
      "    </p>\n",
      "    <p>\n",
      "     Considering the fact that perturbation data are only available for a subset of\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e334\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     perturbed genes, one has to\n",
      "     <italic>\n",
      "      impose\n",
      "     </italic>\n",
      "     a constraint on the orderings between perturbed genes and the rest of the genes in the network. A natural constraint is to assume that no edges exist from unperturbed genes to the perturbed ones. This assumption defines a clear choice for ordering of nodes in the graph: the set of perturbed genes (say 1 to\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e335\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     ) appear before the unperturbed ones in any ordering of nodes. It is then clear that RIPE can be applied to estimate the regulatory edges in the network by obtaining estimates of the two-layer network using the penalized regression approach in (3). The computational efficiency of the RIPE algorithms facilitates its application to estimation of the entire network regulatory interactions based on limited perturbation data. Estimation of the regulatory network of yeast with\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e336\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     genes and\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e337\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     transcription factors based on 1000 orderings takes less than 22 minutes on a 2.7 GHz Laptop with 6 GB of memory. The resulting estimate includes 134 interactions reported in the BIOGRID dataset (true positives) and a total of 10014 edges. In an experiment similar to those reported above, the number of true positives in 1000 random graphs with the same number of edges and similar 2-layer structure no network with equal or larger true positives was observed (\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e338\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     -value\n",
      "     <inline-formula>\n",
      "      <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e339\" xlink:type=\"simple\"/>\n",
      "     </inline-formula>\n",
      "     ). The distribution of number of true positives in comparison to the number of true positives for the RIPE estimator are shown in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0082393.s010\">\n",
      "      Figure S10\n",
      "     </xref>\n",
      "     .\n",
      "    </p>\n",
      "   </sec>\n",
      "  </sec>\n",
      "  <sec id=\"s4\">\n",
      "   <title>\n",
      "    Discussion\n",
      "   </title>\n",
      "   <p>\n",
      "    The proposed methodology offers several advantages over existing approaches in addressing the key problem of reconstructing of regulatory networks. It relies on a global assessment of causal orderings and employs both perturbation screens and steady state expression data for the reconstruction step that boosts performance. Further, the penalized likelihood method used for estimating the edges exhibits a certain degree of robustness to misspecification of the causal orderings, as observed in\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Shojaie1\">\n",
      "     [13]\n",
      "    </xref>\n",
      "    . As mentioned in the introductory section, highly accurate perturbation data may be sufficient for the reconstruction task at hand, but this is not often the case. On the other hand, integrating two data sources proves beneficial, as our numerical work illustrates. We discuss next several issues related to the RIPE algorithm and outline some future research directions.\n",
      "   </p>\n",
      "   <p>\n",
      "    As previously indicated, scalability issues are important to the proposed methodology, since the influence matrix\n",
      "    <inline-formula>\n",
      "     <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e340\" xlink:type=\"simple\"/>\n",
      "    </inline-formula>\n",
      "    usually contains cycles due to natural feedback loops in gene mechanisms and the noisy measurements in the perturbation experiments. Hence, calculating\n",
      "    <italic>\n",
      "     all\n",
      "    </italic>\n",
      "    possible causal orderings compatible with\n",
      "    <inline-formula>\n",
      "     <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e341\" xlink:type=\"simple\"/>\n",
      "    </inline-formula>\n",
      "    may become infeasible given the exponential complexity of the problem. The proposed MC-DFS heuristic offers a fast, reliable alternative. Our numerical experiments suggest that in practice it is not required to exhaustively search the space of possible orderings, and a moderate number of randomly generated orderings often produce comparable estimates. Based on the results reported here, for graphs of up to\n",
      "    <inline-formula>\n",
      "     <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e342\" xlink:type=\"simple\"/>\n",
      "    </inline-formula>\n",
      "    nodes, a total of\n",
      "    <inline-formula>\n",
      "     <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e343\" xlink:type=\"simple\"/>\n",
      "    </inline-formula>\n",
      "    orderings results in reliable estimates.\n",
      "   </p>\n",
      "   <p>\n",
      "    Our extensive evaluation studies strongly suggest that the RIPE algorithm is especially suitable in settings where one deals with noisy perturbation data and fairly good quality steady state expression data are available. Algorithms utilizing only information from perturbation screens work well for topologies without many cycles, while those relying only on observational data are not particularly competitive. Further, other data sources that can further filter the influence matrix, such as binding experiments (e.g. based on ChIP-chip technology), would be beneficial.\n",
      "   </p>\n",
      "   <p>\n",
      "    The proposed methodology is in principle applicable to other organisms for which steady state gene expression data with large sample size exist, such as Arabidopsis, mouse and human. The bottleneck would be the paucity of systematic perturbation screens, since they are more costly to produce. But the RIPE algorithm would be well suited for estimating a regulatory subnetwork for which adequate perturbation data are available.\n",
      "   </p>\n",
      "   <p>\n",
      "    An interesting extension of the proposed methodology involves perturbation screens from time course data\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Frhlich1\">\n",
      "     [8]\n",
      "    </xref>\n",
      "    . The RIPE algorithm can be extended as follows. Relying on the fact that the bperturbation screens convey the causal ordering of the genes in the network, the DAG scoring method itself can be extended to cover time course steady state expression data. For example, a modified version of the likelihood scoring of DAGs can be used to shrink the estimated networks in each time point towards a common skeleton for the underlying network (similar to the approach described in\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Guo1\">\n",
      "     [43]\n",
      "    </xref>\n",
      "    ), or each time point can be modeled as a modified version of the network at the previous time point\n",
      "    <xref ref-type=\"bibr\" rid=\"pone.0082393-Song1\">\n",
      "     [44]\n",
      "    </xref>\n",
      "    .\n",
      "   </p>\n",
      "  </sec>\n",
      "  <sec id=\"s5\">\n",
      "   <title>\n",
      "    Supporting Information\n",
      "   </title>\n",
      "   <supplementary-material id=\"pone.0082393.s001\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s001\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S1\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Small cyclic subnetwork example.\n",
      "      </bold>\n",
      "      The true network (left) includes a number of cycles, and the estimate from the RIPE algorithm correctly identifies some of these cycles (right).\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s002\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s002\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S2\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Numerical study on choices of\n",
      "      </bold>\n",
      "      <bold>\n",
      "       <italic>\n",
      "        τ\n",
      "       </italic>\n",
      "      </bold>\n",
      "      <bold>\n",
      "       and\n",
      "      </bold>\n",
      "      <bold>\n",
      "       <italic>\n",
      "        q\n",
      "       </italic>\n",
      "      </bold>\n",
      "      <bold>\n",
      "       .\n",
      "      </bold>\n",
      "      Values of the Precision (P) for different combinations of\n",
      "      <italic>\n",
      "       τ\n",
      "      </italic>\n",
      "      (threshold for including an edge in the consensus graph) and\n",
      "      <italic>\n",
      "       q\n",
      "      </italic>\n",
      "      (proportion of highest values of the log-likelihood function used in constructing the consensus graph).\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s003\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s003\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S3\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Numerical study on choices of\n",
      "      </bold>\n",
      "      <bold>\n",
      "       <italic>\n",
      "        τ\n",
      "       </italic>\n",
      "      </bold>\n",
      "      <bold>\n",
      "       and\n",
      "      </bold>\n",
      "      <bold>\n",
      "       <italic>\n",
      "        q\n",
      "       </italic>\n",
      "      </bold>\n",
      "      <bold>\n",
      "       .\n",
      "      </bold>\n",
      "      Values of the Recall (\n",
      "      <italic>\n",
      "       R\n",
      "      </italic>\n",
      "      ) for different combinations of\n",
      "      <italic>\n",
      "       τ\n",
      "      </italic>\n",
      "      (threshold for including an edge in the consensus graph) and\n",
      "      <italic>\n",
      "       q\n",
      "      </italic>\n",
      "      proportion of highest values of the log-likelihood function used in constructing the consensus graph).\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s004\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s004\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S4\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Numerical study on choices of\n",
      "      </bold>\n",
      "      <bold>\n",
      "       <italic>\n",
      "        τ\n",
      "       </italic>\n",
      "      </bold>\n",
      "      <bold>\n",
      "       and\n",
      "      </bold>\n",
      "      <bold>\n",
      "       <italic>\n",
      "        q\n",
      "       </italic>\n",
      "      </bold>\n",
      "      <bold>\n",
      "       .\n",
      "      </bold>\n",
      "      Values of the\n",
      "      <italic>\n",
      "       F\n",
      "      </italic>\n",
      "      <sub>\n",
      "       1\n",
      "      </sub>\n",
      "      measure for different combinations of\n",
      "      <italic>\n",
      "       τ\n",
      "      </italic>\n",
      "      (threshold for including an edge in the consensus graph) and\n",
      "      <italic>\n",
      "       q\n",
      "      </italic>\n",
      "      (proportion of highest values of the log-likelihood function used in constructing the consensus graph).\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s005\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s005\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S5\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Influence graph characteristics versus p-value for knockdown experiments in DREAM4.\n",
      "      </bold>\n",
      "      Number of edges (open triangles) in the influence graph and the size of the largest connected component (dots) versus cut-off p-value for differential expression. The data is based on five replications of the knockdown and wildtype experiments for (A) 100-node network 1, (B) 100-node network 3, and (C) 100-node network 5 in the DREAM4 challenge. P-values chosen for analysis were 0.03 for (A) and (C), and 0.027 for (B).\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s006\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s006\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S6\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Synthetic regulatory network.\n",
      "      </bold>\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s007\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s007\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S7\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Illustration of influence matrices for synthetic networks.\n",
      "      </bold>\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e344\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      <sub>\n",
      "       0\n",
      "      </sub>\n",
      "      : ground truth,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e345\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      <sub>\n",
      "       1\n",
      "      </sub>\n",
      "      : 5% of directions reversed,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e346\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      : 10% new effects added,\n",
      "      <inline-formula>\n",
      "       <inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e347\" xlink:type=\"simple\"/>\n",
      "      </inline-formula>\n",
      "      <sub>\n",
      "       3\n",
      "      </sub>\n",
      "      : 5% directions reversed and 10% new effects added. A black dot in position (\n",
      "      <italic>\n",
      "       i\n",
      "      </italic>\n",
      "      ,\n",
      "      <italic>\n",
      "       j\n",
      "      </italic>\n",
      "      ) (i.e., in row i and column j) represents that gene\n",
      "      <italic>\n",
      "       i\n",
      "      </italic>\n",
      "      inuences gene\n",
      "      <italic>\n",
      "       j\n",
      "      </italic>\n",
      "      .\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s008\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s008\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S8\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Effect of increasing number of ordering used in RIPE.\n",
      "      </bold>\n",
      "      The values of\n",
      "      <italic>\n",
      "       F\n",
      "      </italic>\n",
      "      <sub>\n",
      "       1\n",
      "      </sub>\n",
      "      ,\n",
      "      <italic>\n",
      "       P\n",
      "      </italic>\n",
      "      and\n",
      "      <italic>\n",
      "       R\n",
      "      </italic>\n",
      "      are displayed for increasing number of orders in the synthetic DAG with\n",
      "      <italic>\n",
      "       p\n",
      "      </italic>\n",
      "      = 100 nodes under 1.5% false positive edges.\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s009\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s009\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S9\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Influence graph characteristics versus p-value for the yeast regulatory network.\n",
      "      </bold>\n",
      "      Number of edges (open triangles) in the influence graph and the size of the largest connected component (dots) versus cut-off p-value for differential expression in the transcription factor knockout experiments in yeast. The graph is based on expression data from 588 two-color microarrays for 269 transcription factors. The p-value cut-off was chosen to 0.002.\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s010\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s010\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S10\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Performance of RIPE in estimating the layered yeast regulatory network.\n",
      "      </bold>\n",
      "      The distribution of number of true positives, in comparison to the number of true positives for the RIPE estimator, for 1000 random graphs with the same number of edges and similar 2-layer structure (\n",
      "      <italic>\n",
      "       p\n",
      "      </italic>\n",
      "      = 6051 genes and\n",
      "      <italic>\n",
      "       k\n",
      "      </italic>\n",
      "      = 269 transcription factors). No random network with equal or larger true positives was observed (p-value&lt;0∶001).\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s011\" mimetype=\"image/tiff\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s011\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Figure S11\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Simple graphs with 3 nodes.\n",
      "      </bold>\n",
      "     </p>\n",
      "     <p>\n",
      "      (TIFF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "   <supplementary-material id=\"pone.0082393.s012\" mimetype=\"application/pdf\" position=\"float\" xlink:href=\"info:doi/10.1371/journal.pone.0082393.s012\" xlink:type=\"simple\">\n",
      "    <label>\n",
      "     Text S1\n",
      "    </label>\n",
      "    <caption>\n",
      "     <p>\n",
      "      <bold>\n",
      "       Supporting information.\n",
      "      </bold>\n",
      "     </p>\n",
      "     <p>\n",
      "      (PDF)\n",
      "     </p>\n",
      "    </caption>\n",
      "   </supplementary-material>\n",
      "  </sec>\n",
      " </body>\n",
      " <back>\n",
      "  <ack>\n",
      "   <p>\n",
      "    The work was supported by NIH grants 1RC1CA145444-01 to GM and 1R21GM101719-01A1 to GM and AS. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript. The authors would like to thank Dr. Pinna for making available his code for the FFLDR method.\n",
      "   </p>\n",
      "  </ack>\n",
      "  <ref-list>\n",
      "   <title>\n",
      "    References\n",
      "   </title>\n",
      "   <ref id=\"pone.0082393-Markowetz1\">\n",
      "    <label>\n",
      "     1\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Markowetz\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kostka\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Troyanskaya\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       OG\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Spang\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2007\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Nested effects models for highdimensional phenotyping screens\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      23\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      i305\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      i312\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Tresch1\">\n",
      "    <label>\n",
      "     2\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Tresch\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Markowetz\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2008\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Structure learning in Nested Effects Models\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Statistical Applications in Genetics and Molecular Biology\n",
      "     </source>\n",
      "     <volume>\n",
      "      7\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      9\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Klamt1\">\n",
      "    <label>\n",
      "     3\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Klamt\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Flassig\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sundmacher\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       K\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Transwesd: inferring cellular networks with transitive reduction\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      26\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      2160\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      2168\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Piano1\">\n",
      "    <label>\n",
      "     4\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Piano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Schetter\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Morton\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Gunsalus\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       K\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Reinke\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       V\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2002\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Gene clustering based on RNAi phenotypes of ovary-enriched genes in C. elegans\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Current Biology\n",
      "     </source>\n",
      "     <volume>\n",
      "      12\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      1959\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      1964\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Ohya1\">\n",
      "    <label>\n",
      "     5\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Ohya\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Y\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sese\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Yukawa\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Nakatani\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Y\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2005\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      High-dimensional and large-scale phenotyping of yeast mutants\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Proc Natl Acad Sci USA\n",
      "     </source>\n",
      "     <volume>\n",
      "      102\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      19015\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      19020\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Markowetz2\">\n",
      "    <label>\n",
      "     6\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Markowetz\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Bloch\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Spang\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2005\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Non-transcriptional pathway features reconstructed from secondary effects of rna interference\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      21\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      4026\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      4032\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Anchang1\">\n",
      "    <label>\n",
      "     7\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Anchang\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       B\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sadeh\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Jacob\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Tresch\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Vlad\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2009\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Modeling the temporal interplay of molecular signaling and gene expression by using dynamic nested effects models\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Proceedings of the National Academy of Sciences\n",
      "     </source>\n",
      "     <volume>\n",
      "      106\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      6447\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      52\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Frhlich1\">\n",
      "    <label>\n",
      "     8\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Fröhlich\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       H\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Praveen\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       P\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Tresch\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Fast and Efficient Dynamic Nested Effects Models\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <fpage>\n",
      "      238\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      44\n",
      "     </lpage>\n",
      "     <comment>\n",
      "      doi:\n",
      "      <ext-link ext-link-type=\"uri\" xlink:href=\"http://dx.doi.org/10.1093/bioinformatics/btq631\" xlink:type=\"simple\">\n",
      "       10.1093/bioinformatics/btq631\n",
      "      </ext-link>\n",
      "     </comment>\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Pinna1\">\n",
      "    <label>\n",
      "     9\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Pinna\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Soranzo\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       N\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       de la Fuente\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      From knockouts to networks: establishing direct cause-effect relationships through graph analysis\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      PLoS ONE\n",
      "     </source>\n",
      "     <volume>\n",
      "      5\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      e12912\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Greenfield1\">\n",
      "    <label>\n",
      "     10\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Greenfield\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Madar\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Ostrer\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       H\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Bonneau\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      DREAM4: Combining genetic and dynamic information to identify biological networks and dynamical models\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      PLoS ONE\n",
      "     </source>\n",
      "     <volume>\n",
      "      5\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      e13397\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Friedman1\">\n",
      "    <label>\n",
      "     11\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Friedman\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       N\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Koller\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2003\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Being Bayesian about network structure. A Bayesian approach to structure discovery in Bayesian networks\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Machine Learning\n",
      "     </source>\n",
      "     <volume>\n",
      "      50\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      95\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      125\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Pearl1\">\n",
      "    <label>\n",
      "     12\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"book\" xlink:type=\"simple\">\n",
      "     Pearl J (2000) Causality: Models, Reasoning, and Inference. Cambridge: Cambridge University Press.\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Shojaie1\">\n",
      "    <label>\n",
      "     13\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Shojaie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Penalized likelihood methods for estimation of sparse highdimensional directed acyclic graphs\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Biometrika\n",
      "     </source>\n",
      "     <volume>\n",
      "      97\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      519\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      538\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Michailidis1\">\n",
      "    <label>\n",
      "     14\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       d'Alché-Buc\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2013\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Autoregressive models for gene regulatory network inference: Sparsity, stability and causality issues\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Mathematical Biosciences\n",
      "     </source>\n",
      "     <volume>\n",
      "      246\n",
      "     </volume>\n",
      "     (\n",
      "     <issue>\n",
      "      2\n",
      "     </issue>\n",
      "     )\n",
      "     <fpage>\n",
      "      326\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      334\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Maathuis1\">\n",
      "    <label>\n",
      "     15\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Maathuis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       MH\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Colombo\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kalisch\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Bühlmann\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       P\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Predicting causal effects in large-scale systems from observational data\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Nat Methods\n",
      "     </source>\n",
      "     <volume>\n",
      "      7\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      247\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      248\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Spirtes1\">\n",
      "    <label>\n",
      "     16\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"book\" xlink:type=\"simple\">\n",
      "     Spirtes P, Glymour C, Scheines R (2000) Causation, Prediction, and Search. Cambridge, MA: MIT Press.\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Yamaguchi1\">\n",
      "    <label>\n",
      "     17\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Yamaguchi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Yoshida\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Imoto\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Higuchi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       T\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Miyano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2007\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Finding module-based gene networks with state-space models-Mining high-dimensional and short time-course gene expression data\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      IEEE Signal Processing Magazine\n",
      "     </source>\n",
      "     <volume>\n",
      "      24\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      37\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      46\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-OpgenRhein1\">\n",
      "    <label>\n",
      "     18\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Opgen-Rhein\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Strimmer\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       K\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2007\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Learning causal networks from systems biology time course data: an effective model selection procedure for the vector autoregressive process\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      BMC bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      8\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      S3\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Ong1\">\n",
      "    <label>\n",
      "     19\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Ong\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       I\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Glasner\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Page\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2002\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Modelling regulatory pathways in E. coli from time series expression profiles\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      18\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      S241\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      S248\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Perrin1\">\n",
      "    <label>\n",
      "     20\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Perrin\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       B\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Ralaivola\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       L\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Mazurie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Bottani\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Mallet\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2003\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Gene networks inference using dynamic Bayesian networks\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      19\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      138\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      148\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Fujita1\">\n",
      "    <label>\n",
      "     21\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Fujita\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sato\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Garay-Malpartida\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       H\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Yamaguchi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Miyano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2007\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Modeling gene expression regulatory networks with the sparse vector autoregressive model\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      BMC Systems Biology\n",
      "     </source>\n",
      "     <volume>\n",
      "      1\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      39\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Mukhopadhyay1\">\n",
      "    <label>\n",
      "     22\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Mukhopadhyay\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       N\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Chatterjee\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2007\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Causality and pathway search in microarray time series experiment\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      23\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      442\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Shojaie2\">\n",
      "    <label>\n",
      "     23\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Shojaie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Discovering graphical Granger causality using the truncating lasso penalty\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      26\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      i517\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Shojaie3\">\n",
      "    <label>\n",
      "     24\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Shojaie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Basu\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2012\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Adaptive thresholding for reconstructing regulatory networks from time-course gene expression data\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Statistics in Biosciences\n",
      "     </source>\n",
      "     <volume>\n",
      "      4\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      66\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      83\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Stelling1\">\n",
      "    <label>\n",
      "     25\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Stelling\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sauer\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       U\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Szallasi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Z\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Doyle\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       F\n",
      "      </given-names>\n",
      "      <suffix>\n",
      "       III\n",
      "      </suffix>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Doyle\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2004\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Robustness of cellular functions\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Cell\n",
      "     </source>\n",
      "     <volume>\n",
      "      118\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      675\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      685\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Kitano1\">\n",
      "    <label>\n",
      "     26\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kitano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       H\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2004\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Biological robustness\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Nature Reviews Genetics\n",
      "     </source>\n",
      "     <volume>\n",
      "      5\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      826\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      837\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Dasgupta1\">\n",
      "    <label>\n",
      "     27\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"book\" xlink:type=\"simple\">\n",
      "     Dasgupta S, Papadimitriou CH, Vazirani U (2008) Algorithms, 1st edition. New York: McGraw-Hill, Inc.\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Tarjan1\">\n",
      "    <label>\n",
      "     28\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Tarjan\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       R\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      1972\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Depth-First Search and Linear Graph Algorithms\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      SIAM Journal on Computing\n",
      "     </source>\n",
      "     <volume>\n",
      "      1\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      146\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      160\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Knuth1\">\n",
      "    <label>\n",
      "     29\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Knuth\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       DE\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Szwarcfiter\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       JL\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      1974\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      A structured program to generate all topological sorting arrangements\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Inf Process Lett\n",
      "     </source>\n",
      "     <volume>\n",
      "      2\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      153\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      157\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Stark1\">\n",
      "    <label>\n",
      "     30\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Stark\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       C\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Breitkreutz\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       BJ\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Reguly\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       T\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Boucher\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       L\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Breitkreutz\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2006\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      BioGRID: a general repository for interaction datasets\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Nucleic Acids Res\n",
      "     </source>\n",
      "     <volume>\n",
      "      34\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      D535\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      539\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Csardi1\">\n",
      "    <label>\n",
      "     31\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Csardi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Nepusz\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       T\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2006\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      The igraph Software Package for Complex Network Research\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Inter-Journal Complex Systems\n",
      "     </source>\n",
      "     <fpage>\n",
      "      1695\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Prill1\">\n",
      "    <label>\n",
      "     32\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Prill\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       RJ\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Marbach\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Saez-Rodriguez\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Sorger\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       PK\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Alexopoulos\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       LG\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Towards a rigorous assessment of systems biology models: the DREAM3 challenges\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      PLoS ONE\n",
      "     </source>\n",
      "     <volume>\n",
      "      5\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      e9202\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Marbach1\">\n",
      "    <label>\n",
      "     33\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Marbach\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Schaffter\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       T\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Mattiussi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       C\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Floreano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2009\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Generating realistic in silico gene networks for performance assessment of reverse engineering methods\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      J Comput Biol\n",
      "     </source>\n",
      "     <volume>\n",
      "      16\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      229\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      239\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Marbach2\">\n",
      "    <label>\n",
      "     34\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Marbach\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Prill\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       RJ\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Schaffter\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       T\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Mattiussi\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       C\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Floreano\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Revealing strengths and weaknesses of methods for gene network inference\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Proceedings of the National Academy of Sciences\n",
      "     </source>\n",
      "     <volume>\n",
      "      107\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      6286\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      6291\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Shojaie4\">\n",
      "    <label>\n",
      "     35\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Shojaie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Network enrichment analysis in complex experiments\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Statistical Applications in Genetics and Molecular Biology\n",
      "     </source>\n",
      "     <volume>\n",
      "      9\n",
      "     </volume>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Margolin1\">\n",
      "    <label>\n",
      "     36\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Margolin\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Nemenman\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       I\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Basso\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       K\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Wiggins\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       C\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Stolovitzky\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2006\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      ARACNE: an algorithm for the reconstruction of gene regulatory networks in a mammalian cellular context\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      BMC bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      7\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      S7\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Kalisch1\">\n",
      "    <label>\n",
      "     37\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kalisch\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Mächler\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Colombo\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       D\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Maathuis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       MH\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Bühlmann\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       P\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2012\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Causal inference using graphical models with the R package pcalg\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Journal of Statistical Software\n",
      "     </source>\n",
      "     <volume>\n",
      "      47\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      1\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      26\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Shojaie5\">\n",
      "    <label>\n",
      "     38\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Shojaie\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       A\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2009\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Analysis of Gene Sets Based on the Underlying Regulatory Network\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Journal of Computational Biology\n",
      "     </source>\n",
      "     <volume>\n",
      "      16\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      407\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      426\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Hu1\">\n",
      "    <label>\n",
      "     39\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Hu\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Z\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Killion\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       PJ\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Iyer\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       VR\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2007\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Genetic reconstruction of a functional transcriptional regulatory network\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Nat Genet\n",
      "     </source>\n",
      "     <volume>\n",
      "      39\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      683\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      687\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Reimand1\">\n",
      "    <label>\n",
      "     40\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Reimand\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Vaquerizas\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       JM\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Todd\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       AE\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Vilo\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Luscombe\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       NM\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Comprehensive reanalysis of transcription factor knockout expression data in Saccharomyces cerevisiae reveals many new targets\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Nucleic Acids Res\n",
      "     </source>\n",
      "     <volume>\n",
      "      38\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      4768\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      4777\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Benjamini1\">\n",
      "    <label>\n",
      "     41\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Benjamini\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Y\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Hochberg\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       Y\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      1995\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Controlling the false discovery rate: a practical and powerful approach to multiple testing\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      J Roy Statist Soc Ser B\n",
      "     </source>\n",
      "     <volume>\n",
      "      57\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      289\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      300\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-vanWageningen1\">\n",
      "    <label>\n",
      "     42\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       van Wageningen\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       S\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kemmeren\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       P\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Lijnzaad\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       P\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Margaritis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       T\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Benschop\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       JJ\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <etal>\n",
      "      et al\n",
      "     </etal>\n",
      "     . (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Functional overlap and regulatory links shape genetic interactions between signaling pathways\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Cell\n",
      "     </source>\n",
      "     <volume>\n",
      "      143\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      991\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      1004\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Guo1\">\n",
      "    <label>\n",
      "     43\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Guo\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Levina\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       E\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Michailidis\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       G\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Zhu\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       J\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2010\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      Joint estimation of multiple graphical models\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Biometrika\n",
      "     </source>\n",
      "     <volume>\n",
      "      98\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      1\n",
      "     </fpage>\n",
      "     –\n",
      "     <lpage>\n",
      "      16\n",
      "     </lpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "   <ref id=\"pone.0082393-Song1\">\n",
      "    <label>\n",
      "     44\n",
      "    </label>\n",
      "    <mixed-citation publication-type=\"journal\" xlink:type=\"simple\">\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Song\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       L\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Kolar\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       M\n",
      "      </given-names>\n",
      "     </name>\n",
      "     ,\n",
      "     <name name-style=\"western\">\n",
      "      <surname>\n",
      "       Xing\n",
      "      </surname>\n",
      "      <given-names>\n",
      "       E\n",
      "      </given-names>\n",
      "     </name>\n",
      "     (\n",
      "     <year>\n",
      "      2009\n",
      "     </year>\n",
      "     )\n",
      "     <article-title>\n",
      "      KELLER: estimating time-varying interactions between genes\n",
      "     </article-title>\n",
      "     .\n",
      "     <source>\n",
      "      Bioinformatics\n",
      "     </source>\n",
      "     <volume>\n",
      "      25\n",
      "     </volume>\n",
      "     :\n",
      "     <fpage>\n",
      "      i128\n",
      "     </fpage>\n",
      "     .\n",
      "    </mixed-citation>\n",
      "   </ref>\n",
      "  </ref-list>\n",
      " </back>\n",
      "</article>\n",
      "Test extracting abstract\n",
      "\n",
      "Reconstructing transcriptional regulatory networks is an important task in functional genomics. Data obtained from experiments that perturb genes by knockouts or RNA interference contain useful information for addressing this reconstruction problem. However, such data can be limited in size and/or are expensive to acquire. On the other hand, observational data of the organism in steady state (e.g., wild-type) are more readily available, but their informational content is inadequate for the task at hand. We develop a computational approach to appropriately utilize both data sources for estimating a regulatory network. The proposed approach is based on a three-step algorithm to estimate the underlying directed but cyclic network, that uses as input both perturbation screens and steady state gene expression data. In the first step, the algorithm determines causal orderings of the genes that are consistent with the perturbation data, by combining an exhaustive search method with a fast heuristic that in turn couples a Monte Carlo technique with a fast search algorithm. In the second step, for each obtained causal ordering, a regulatory network is estimated using a penalized likelihood based method, while in the third step a consensus network is constructed from the highest scored ones. Extensive computational experiments show that the algorithm performs well in reconstructing the underlying network and clearly outperforms competing approaches that rely only on a single data source. Further, it is established that the algorithm produces a consistent estimate of the regulatory network.\n",
      "\n",
      "Test extracting references\n",
      "<title>Introduction</title>\n",
      "<title>Overview and details of the RIPE method.</title>\n",
      "<title>Methods</title>\n",
      "<title>Method Overview</title>\n",
      "<title>Obtaining Causal Orderings from Perturbation Data</title>\n",
      "<title>Estimating the Influence Matrix from Perturbation Data</title>\n",
      "<title>Obtaining Causal Orderings based on the Influence Matrix</title>\n",
      "<title>Algorithm for DFS.</title>\n",
      "<title>Estimation of Network Structure Using Gene Expression Data</title>\n",
      "<title>Graph averaging: a consensus regulatory network</title>\n",
      "<title>Choice of Parameters and Properties of the Algorithm</title>\n",
      "<title>Results</title>\n",
      "<title>Preliminaries</title>\n",
      "<title>Determining the Influence Matrix</title>\n",
      "<title>Influence graph characteristics versus <inline-formula><inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e221\" xlink:type=\"simple\"/></inline-formula>-value for DREAM4.</title>\n",
      "<title>Performance Evaluation Criteria</title>\n",
      "<title>DREAM4 Challenge</title>\n",
      "<title>Reconstruction results for DREAM4 networks with simulated steady state data.</title>\n",
      "<title>Reconstruction results for DREAM4 networks with multifactorial data.</title>\n",
      "<title>Experiments with Synthetic Data</title>\n",
      "<title>Small Directed Acyclic Graph</title>\n",
      "<title>Performance of RIPE and competing methods on the reconstruction of synthetic networks.</title>\n",
      "<title>Reconstruction results for synthetic networks subject to error in the perturbation data.</title>\n",
      "<title>Impact of increasing number of orderings used in the RIPE algorithm.</title>\n",
      "<title>Effect of False Positive and Negative Errors in Perturbation Data</title>\n",
      "<title>Impact of increasing false positive and negative errors in perturbation data on estimation of acyclic graphs.</title>\n",
      "<title>Large Cyclic Graph (<inline-formula><inline-graphic xlink:href=\"info:doi/10.1371/journal.pone.0082393.e312\" xlink:type=\"simple\"/></inline-formula>)</title>\n",
      "<title>Impact of increasing false positive and negative errors in perturbation data on estimation of high dimensional cyclic graphs.</title>\n",
      "<title>Regulatory Network in Yeast</title>\n",
      "<title>Performance evaluation for the reconstruction of the yeast regulatory network.</title>\n",
      "<title>Discussion</title>\n",
      "<title>Supporting Information</title>\n",
      "<title>References</title>\n"
     ]
    }
   ],
   "source": [
    "# try to read from one of the xml files\n",
    "\n",
    "ref_idx = random.randint(0,L)\n",
    "print('document number {}'.format(ref_idx))\n",
    "\n",
    "path = path2directory + xml_list[ref_idx]\n",
    "print('PATH: {}'.format(path))\n",
    "with open(path) as f:\n",
    "    xml_soup = BeautifulSoup(f,'xml')\n",
    "    print('CONTENTS:')\n",
    "    print(xml_soup.prettify())\n",
    "\n",
    "    print('Test extracting abstract')\n",
    "    titles = xml_soup.find_all('abstract')\n",
    "    for title in titles:\n",
    "        print(title.text)\n",
    "        \n",
    "    print('Test extracting publication year')\n",
    "    titles = xml_soup.find_all('title')\n",
    "    for title in titles:\n",
    "        print(title)\n",
    "        \n",
    "    print('Test extracting authors')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sandbox (temp)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first steps:\n",
    "\n",
    "# parse the title\n",
    "\n",
    "# parse the abstract\n",
    "\n",
    "# parse the publication year\n",
    "\n",
    "\n",
    "\n",
    "# next steps:\n",
    "\n",
    "# parse the authors\n",
    "\n",
    "# parse the introduction\n",
    "\n",
    "# grab the methods section\n",
    "\n",
    "# grab the results\n",
    "\n",
    "# grab the discussion\n",
    "\n",
    "# grab the references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
